// @aws-sdk/client-mediaconvert/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooManyRequestsException = exports.ServiceQuotaExceededException = exports.NotFoundException = exports.InternalServerErrorException = exports.ForbiddenException = exports.ConflictException = exports.BadRequestException = void 0;
const MediaConvertServiceException_1 = require("./MediaConvertServiceException");
class BadRequestException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "BadRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
        this.Message = opts.Message;
    }
}
exports.BadRequestException = BadRequestException;
class ConflictException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConflictException = ConflictException;
class ForbiddenException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "ForbiddenException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ForbiddenException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ForbiddenException.prototype);
        this.Message = opts.Message;
    }
}
exports.ForbiddenException = ForbiddenException;
class InternalServerErrorException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "InternalServerErrorException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServerErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerErrorException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerErrorException = InternalServerErrorException;
class NotFoundException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "NotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.NotFoundException = NotFoundException;
class ServiceQuotaExceededException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
exports.ServiceQuotaExceededException = ServiceQuotaExceededException;
class TooManyRequestsException extends MediaConvertServiceException_1.MediaConvertServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
