// @aws-sdk/client-mediaconvert/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var MediaConvertServiceException = require('./models/MediaConvertServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "mediaconvert",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class MediaConvertClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultMediaConvertHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AssociateCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "AssociateCertificate", {})
    .n("MediaConvertClient", "AssociateCertificateCommand")
    .sc(schemas_0.AssociateCertificate$)
    .build() {
}

class CancelJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CancelJob", {})
    .n("MediaConvertClient", "CancelJobCommand")
    .sc(schemas_0.CancelJob$)
    .build() {
}

class CreateJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CreateJob", {})
    .n("MediaConvertClient", "CreateJobCommand")
    .sc(schemas_0.CreateJob$)
    .build() {
}

class CreateJobTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CreateJobTemplate", {})
    .n("MediaConvertClient", "CreateJobTemplateCommand")
    .sc(schemas_0.CreateJobTemplate$)
    .build() {
}

class CreatePresetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CreatePreset", {})
    .n("MediaConvertClient", "CreatePresetCommand")
    .sc(schemas_0.CreatePreset$)
    .build() {
}

class CreateQueueCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CreateQueue", {})
    .n("MediaConvertClient", "CreateQueueCommand")
    .sc(schemas_0.CreateQueue$)
    .build() {
}

class CreateResourceShareCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "CreateResourceShare", {})
    .n("MediaConvertClient", "CreateResourceShareCommand")
    .sc(schemas_0.CreateResourceShare$)
    .build() {
}

class DeleteJobTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DeleteJobTemplate", {})
    .n("MediaConvertClient", "DeleteJobTemplateCommand")
    .sc(schemas_0.DeleteJobTemplate$)
    .build() {
}

class DeletePolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DeletePolicy", {})
    .n("MediaConvertClient", "DeletePolicyCommand")
    .sc(schemas_0.DeletePolicy$)
    .build() {
}

class DeletePresetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DeletePreset", {})
    .n("MediaConvertClient", "DeletePresetCommand")
    .sc(schemas_0.DeletePreset$)
    .build() {
}

class DeleteQueueCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DeleteQueue", {})
    .n("MediaConvertClient", "DeleteQueueCommand")
    .sc(schemas_0.DeleteQueue$)
    .build() {
}

class DescribeEndpointsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DescribeEndpoints", {})
    .n("MediaConvertClient", "DescribeEndpointsCommand")
    .sc(schemas_0.DescribeEndpoints$)
    .build() {
}

class DisassociateCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "DisassociateCertificate", {})
    .n("MediaConvertClient", "DisassociateCertificateCommand")
    .sc(schemas_0.DisassociateCertificate$)
    .build() {
}

class GetJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetJob", {})
    .n("MediaConvertClient", "GetJobCommand")
    .sc(schemas_0.GetJob$)
    .build() {
}

class GetJobsQueryResultsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetJobsQueryResults", {})
    .n("MediaConvertClient", "GetJobsQueryResultsCommand")
    .sc(schemas_0.GetJobsQueryResults$)
    .build() {
}

class GetJobTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetJobTemplate", {})
    .n("MediaConvertClient", "GetJobTemplateCommand")
    .sc(schemas_0.GetJobTemplate$)
    .build() {
}

class GetPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetPolicy", {})
    .n("MediaConvertClient", "GetPolicyCommand")
    .sc(schemas_0.GetPolicy$)
    .build() {
}

class GetPresetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetPreset", {})
    .n("MediaConvertClient", "GetPresetCommand")
    .sc(schemas_0.GetPreset$)
    .build() {
}

class GetQueueCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "GetQueue", {})
    .n("MediaConvertClient", "GetQueueCommand")
    .sc(schemas_0.GetQueue$)
    .build() {
}

class ListJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListJobs", {})
    .n("MediaConvertClient", "ListJobsCommand")
    .sc(schemas_0.ListJobs$)
    .build() {
}

class ListJobTemplatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListJobTemplates", {})
    .n("MediaConvertClient", "ListJobTemplatesCommand")
    .sc(schemas_0.ListJobTemplates$)
    .build() {
}

class ListPresetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListPresets", {})
    .n("MediaConvertClient", "ListPresetsCommand")
    .sc(schemas_0.ListPresets$)
    .build() {
}

class ListQueuesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListQueues", {})
    .n("MediaConvertClient", "ListQueuesCommand")
    .sc(schemas_0.ListQueues$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListTagsForResource", {})
    .n("MediaConvertClient", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "ListVersions", {})
    .n("MediaConvertClient", "ListVersionsCommand")
    .sc(schemas_0.ListVersions$)
    .build() {
}

class ProbeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "Probe", {})
    .n("MediaConvertClient", "ProbeCommand")
    .sc(schemas_0.Probe$)
    .build() {
}

class PutPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "PutPolicy", {})
    .n("MediaConvertClient", "PutPolicyCommand")
    .sc(schemas_0.PutPolicy$)
    .build() {
}

class SearchJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "SearchJobs", {})
    .n("MediaConvertClient", "SearchJobsCommand")
    .sc(schemas_0.SearchJobs$)
    .build() {
}

class StartJobsQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "StartJobsQuery", {})
    .n("MediaConvertClient", "StartJobsQueryCommand")
    .sc(schemas_0.StartJobsQuery$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "TagResource", {})
    .n("MediaConvertClient", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "UntagResource", {})
    .n("MediaConvertClient", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateJobTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "UpdateJobTemplate", {})
    .n("MediaConvertClient", "UpdateJobTemplateCommand")
    .sc(schemas_0.UpdateJobTemplate$)
    .build() {
}

class UpdatePresetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "UpdatePreset", {})
    .n("MediaConvertClient", "UpdatePresetCommand")
    .sc(schemas_0.UpdatePreset$)
    .build() {
}

class UpdateQueueCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("MediaConvert", "UpdateQueue", {})
    .n("MediaConvertClient", "UpdateQueueCommand")
    .sc(schemas_0.UpdateQueue$)
    .build() {
}

const paginateDescribeEndpoints = core.createPaginator(MediaConvertClient, DescribeEndpointsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListJobs = core.createPaginator(MediaConvertClient, ListJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListJobTemplates = core.createPaginator(MediaConvertClient, ListJobTemplatesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListPresets = core.createPaginator(MediaConvertClient, ListPresetsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListQueues = core.createPaginator(MediaConvertClient, ListQueuesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListVersions = core.createPaginator(MediaConvertClient, ListVersionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateSearchJobs = core.createPaginator(MediaConvertClient, SearchJobsCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AssociateCertificateCommand,
    CancelJobCommand,
    CreateJobCommand,
    CreateJobTemplateCommand,
    CreatePresetCommand,
    CreateQueueCommand,
    CreateResourceShareCommand,
    DeleteJobTemplateCommand,
    DeletePolicyCommand,
    DeletePresetCommand,
    DeleteQueueCommand,
    DescribeEndpointsCommand,
    DisassociateCertificateCommand,
    GetJobCommand,
    GetJobsQueryResultsCommand,
    GetJobTemplateCommand,
    GetPolicyCommand,
    GetPresetCommand,
    GetQueueCommand,
    ListJobsCommand,
    ListJobTemplatesCommand,
    ListPresetsCommand,
    ListQueuesCommand,
    ListTagsForResourceCommand,
    ListVersionsCommand,
    ProbeCommand,
    PutPolicyCommand,
    SearchJobsCommand,
    StartJobsQueryCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateJobTemplateCommand,
    UpdatePresetCommand,
    UpdateQueueCommand,
};
const paginators = {
    paginateDescribeEndpoints,
    paginateListJobs,
    paginateListJobTemplates,
    paginateListPresets,
    paginateListQueues,
    paginateListVersions,
    paginateSearchJobs,
};
class MediaConvert extends MediaConvertClient {
}
smithyClient.createAggregatedClient(commands, MediaConvert, { paginators });

const RequiredFlag = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AudioChannelTag = {
    C: "C",
    CS: "CS",
    HI: "HI",
    L: "L",
    LC: "LC",
    LFE: "LFE",
    LFE2: "LFE2",
    LS: "LS",
    LSD: "LSD",
    LT: "LT",
    LW: "LW",
    M: "M",
    NAR: "NAR",
    R: "R",
    RC: "RC",
    RS: "RS",
    RSD: "RSD",
    RSL: "RSL",
    RSR: "RSR",
    RT: "RT",
    RW: "RW",
    TBC: "TBC",
    TBL: "TBL",
    TBR: "TBR",
    TCS: "TCS",
    VHC: "VHC",
    VHL: "VHL",
    VHR: "VHR",
};
const AudioNormalizationAlgorithm = {
    ITU_BS_1770_1: "ITU_BS_1770_1",
    ITU_BS_1770_2: "ITU_BS_1770_2",
    ITU_BS_1770_3: "ITU_BS_1770_3",
    ITU_BS_1770_4: "ITU_BS_1770_4",
};
const AudioNormalizationAlgorithmControl = {
    CORRECT_AUDIO: "CORRECT_AUDIO",
    MEASURE_ONLY: "MEASURE_ONLY",
};
const AudioNormalizationLoudnessLogging = {
    DONT_LOG: "DONT_LOG",
    LOG: "LOG",
};
const AudioNormalizationPeakCalculation = {
    NONE: "NONE",
    TRUE_PEAK: "TRUE_PEAK",
};
const SlowPalPitchCorrection = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AudioTypeControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
const AacAudioDescriptionBroadcasterMix = {
    BROADCASTER_MIXED_AD: "BROADCASTER_MIXED_AD",
    NORMAL: "NORMAL",
};
const AacCodecProfile = {
    HEV1: "HEV1",
    HEV2: "HEV2",
    LC: "LC",
    XHE: "XHE",
};
const AacCodingMode = {
    AD_RECEIVER_MIX: "AD_RECEIVER_MIX",
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_1_1: "CODING_MODE_1_1",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_5_1: "CODING_MODE_5_1",
    CODING_MODE_AUTO: "CODING_MODE_AUTO",
};
const AacLoudnessMeasurementMode = {
    ANCHOR: "ANCHOR",
    PROGRAM: "PROGRAM",
};
const AacRateControlMode = {
    CBR: "CBR",
    VBR: "VBR",
};
const AacRawFormat = {
    LATM_LOAS: "LATM_LOAS",
    NONE: "NONE",
};
const AacSpecification = {
    MPEG2: "MPEG2",
    MPEG4: "MPEG4",
};
const AacVbrQuality = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM_HIGH: "MEDIUM_HIGH",
    MEDIUM_LOW: "MEDIUM_LOW",
};
const Ac3BitstreamMode = {
    COMMENTARY: "COMMENTARY",
    COMPLETE_MAIN: "COMPLETE_MAIN",
    DIALOGUE: "DIALOGUE",
    EMERGENCY: "EMERGENCY",
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    MUSIC_AND_EFFECTS: "MUSIC_AND_EFFECTS",
    VISUALLY_IMPAIRED: "VISUALLY_IMPAIRED",
    VOICE_OVER: "VOICE_OVER",
};
const Ac3CodingMode = {
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_1_1: "CODING_MODE_1_1",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_3_2_LFE: "CODING_MODE_3_2_LFE",
    CODING_MODE_AUTO: "CODING_MODE_AUTO",
};
const Ac3DynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Ac3DynamicRangeCompressionProfile = {
    FILM_STANDARD: "FILM_STANDARD",
    NONE: "NONE",
};
const Ac3DynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Ac3LfeFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Ac3MetadataControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
const AudioCodec = {
    AAC: "AAC",
    AC3: "AC3",
    AIFF: "AIFF",
    EAC3: "EAC3",
    EAC3_ATMOS: "EAC3_ATMOS",
    FLAC: "FLAC",
    MP2: "MP2",
    MP3: "MP3",
    OPUS: "OPUS",
    PASSTHROUGH: "PASSTHROUGH",
    VORBIS: "VORBIS",
    WAV: "WAV",
};
const Eac3AtmosBitstreamMode = {
    COMPLETE_MAIN: "COMPLETE_MAIN",
};
const Eac3AtmosCodingMode = {
    CODING_MODE_5_1_4: "CODING_MODE_5_1_4",
    CODING_MODE_7_1_4: "CODING_MODE_7_1_4",
    CODING_MODE_9_1_6: "CODING_MODE_9_1_6",
    CODING_MODE_AUTO: "CODING_MODE_AUTO",
};
const Eac3AtmosDialogueIntelligence = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Eac3AtmosDownmixControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Eac3AtmosDynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Eac3AtmosDynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Eac3AtmosDynamicRangeControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Eac3AtmosMeteringMode = {
    ITU_BS_1770_1: "ITU_BS_1770_1",
    ITU_BS_1770_2: "ITU_BS_1770_2",
    ITU_BS_1770_3: "ITU_BS_1770_3",
    ITU_BS_1770_4: "ITU_BS_1770_4",
    LEQ_A: "LEQ_A",
};
const Eac3AtmosStereoDownmix = {
    DPL2: "DPL2",
    NOT_INDICATED: "NOT_INDICATED",
    STEREO: "STEREO",
    SURROUND: "SURROUND",
};
const Eac3AtmosSurroundExMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
const Eac3AttenuationControl = {
    ATTENUATE_3_DB: "ATTENUATE_3_DB",
    NONE: "NONE",
};
const Eac3BitstreamMode = {
    COMMENTARY: "COMMENTARY",
    COMPLETE_MAIN: "COMPLETE_MAIN",
    EMERGENCY: "EMERGENCY",
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    VISUALLY_IMPAIRED: "VISUALLY_IMPAIRED",
};
const Eac3CodingMode = {
    CODING_MODE_1_0: "CODING_MODE_1_0",
    CODING_MODE_2_0: "CODING_MODE_2_0",
    CODING_MODE_3_2: "CODING_MODE_3_2",
    CODING_MODE_AUTO: "CODING_MODE_AUTO",
};
const Eac3DcFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Eac3DynamicRangeCompressionLine = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Eac3DynamicRangeCompressionRf = {
    FILM_LIGHT: "FILM_LIGHT",
    FILM_STANDARD: "FILM_STANDARD",
    MUSIC_LIGHT: "MUSIC_LIGHT",
    MUSIC_STANDARD: "MUSIC_STANDARD",
    NONE: "NONE",
    SPEECH: "SPEECH",
};
const Eac3LfeControl = {
    LFE: "LFE",
    NO_LFE: "NO_LFE",
};
const Eac3LfeFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Eac3MetadataControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
const Eac3PassthroughControl = {
    NO_PASSTHROUGH: "NO_PASSTHROUGH",
    WHEN_POSSIBLE: "WHEN_POSSIBLE",
};
const Eac3PhaseControl = {
    NO_SHIFT: "NO_SHIFT",
    SHIFT_90_DEGREES: "SHIFT_90_DEGREES",
};
const Eac3StereoDownmix = {
    DPL2: "DPL2",
    LO_RO: "LO_RO",
    LT_RT: "LT_RT",
    NOT_INDICATED: "NOT_INDICATED",
};
const Eac3SurroundExMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
const Eac3SurroundMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    NOT_INDICATED: "NOT_INDICATED",
};
const Mp2AudioDescriptionMix = {
    BROADCASTER_MIXED_AD: "BROADCASTER_MIXED_AD",
    NONE: "NONE",
};
const Mp3RateControlMode = {
    CBR: "CBR",
    VBR: "VBR",
};
const WavFormat = {
    EXTENSIBLE: "EXTENSIBLE",
    RF64: "RF64",
    RIFF: "RIFF",
};
const LanguageCode = {
    AAR: "AAR",
    ABK: "ABK",
    AFR: "AFR",
    AKA: "AKA",
    AMH: "AMH",
    ARA: "ARA",
    ARG: "ARG",
    ASM: "ASM",
    AVA: "AVA",
    AVE: "AVE",
    AYM: "AYM",
    AZE: "AZE",
    BAK: "BAK",
    BAM: "BAM",
    BEL: "BEL",
    BEN: "BEN",
    BIH: "BIH",
    BIS: "BIS",
    BOD: "BOD",
    BOS: "BOS",
    BRE: "BRE",
    BUL: "BUL",
    CAT: "CAT",
    CES: "CES",
    CHA: "CHA",
    CHE: "CHE",
    CHU: "CHU",
    CHV: "CHV",
    COR: "COR",
    COS: "COS",
    CRE: "CRE",
    CYM: "CYM",
    DAN: "DAN",
    DEU: "DEU",
    DIV: "DIV",
    DZO: "DZO",
    ELL: "ELL",
    ENG: "ENG",
    ENM: "ENM",
    EPO: "EPO",
    EST: "EST",
    EUS: "EUS",
    EWE: "EWE",
    FAO: "FAO",
    FAS: "FAS",
    FIJ: "FIJ",
    FIN: "FIN",
    FRA: "FRA",
    FRM: "FRM",
    FRY: "FRY",
    FUL: "FUL",
    GER: "GER",
    GLA: "GLA",
    GLE: "GLE",
    GLG: "GLG",
    GLV: "GLV",
    GRN: "GRN",
    GUJ: "GUJ",
    HAT: "HAT",
    HAU: "HAU",
    HEB: "HEB",
    HER: "HER",
    HIN: "HIN",
    HMO: "HMO",
    HRV: "HRV",
    HUN: "HUN",
    HYE: "HYE",
    IBO: "IBO",
    IDO: "IDO",
    III: "III",
    IKU: "IKU",
    ILE: "ILE",
    INA: "INA",
    IND: "IND",
    IPK: "IPK",
    ISL: "ISL",
    ITA: "ITA",
    JAV: "JAV",
    JPN: "JPN",
    KAL: "KAL",
    KAN: "KAN",
    KAS: "KAS",
    KAT: "KAT",
    KAU: "KAU",
    KAZ: "KAZ",
    KHM: "KHM",
    KIK: "KIK",
    KIN: "KIN",
    KIR: "KIR",
    KOM: "KOM",
    KON: "KON",
    KOR: "KOR",
    KUA: "KUA",
    KUR: "KUR",
    LAO: "LAO",
    LAT: "LAT",
    LAV: "LAV",
    LIM: "LIM",
    LIN: "LIN",
    LIT: "LIT",
    LTZ: "LTZ",
    LUB: "LUB",
    LUG: "LUG",
    MAH: "MAH",
    MAL: "MAL",
    MAR: "MAR",
    MKD: "MKD",
    MLG: "MLG",
    MLT: "MLT",
    MON: "MON",
    MRI: "MRI",
    MSA: "MSA",
    MYA: "MYA",
    NAU: "NAU",
    NAV: "NAV",
    NBL: "NBL",
    NDE: "NDE",
    NDO: "NDO",
    NEP: "NEP",
    NLD: "NLD",
    NNO: "NNO",
    NOB: "NOB",
    NOR: "NOR",
    NYA: "NYA",
    OCI: "OCI",
    OJI: "OJI",
    ORI: "ORI",
    ORJ: "ORJ",
    ORM: "ORM",
    OSS: "OSS",
    PAN: "PAN",
    PLI: "PLI",
    POL: "POL",
    POR: "POR",
    PUS: "PUS",
    QAA: "QAA",
    QPC: "QPC",
    QUE: "QUE",
    ROH: "ROH",
    RON: "RON",
    RUN: "RUN",
    RUS: "RUS",
    SAG: "SAG",
    SAN: "SAN",
    SIN: "SIN",
    SLK: "SLK",
    SLV: "SLV",
    SME: "SME",
    SMO: "SMO",
    SNA: "SNA",
    SND: "SND",
    SOM: "SOM",
    SOT: "SOT",
    SPA: "SPA",
    SQI: "SQI",
    SRB: "SRB",
    SRD: "SRD",
    SRP: "SRP",
    SSW: "SSW",
    SUN: "SUN",
    SWA: "SWA",
    SWE: "SWE",
    TAH: "TAH",
    TAM: "TAM",
    TAT: "TAT",
    TEL: "TEL",
    TGK: "TGK",
    TGL: "TGL",
    THA: "THA",
    TIR: "TIR",
    TNG: "TNG",
    TON: "TON",
    TSN: "TSN",
    TSO: "TSO",
    TUK: "TUK",
    TUR: "TUR",
    TWI: "TWI",
    UIG: "UIG",
    UKR: "UKR",
    URD: "URD",
    UZB: "UZB",
    VEN: "VEN",
    VIE: "VIE",
    VOL: "VOL",
    WLN: "WLN",
    WOL: "WOL",
    XHO: "XHO",
    YID: "YID",
    YOR: "YOR",
    ZHA: "ZHA",
    ZHO: "ZHO",
    ZUL: "ZUL",
};
const AudioLanguageCodeControl = {
    FOLLOW_INPUT: "FOLLOW_INPUT",
    USE_CONFIGURED: "USE_CONFIGURED",
};
const RuleType = {
    ALLOWED_RENDITIONS: "ALLOWED_RENDITIONS",
    FORCE_INCLUDE_RENDITIONS: "FORCE_INCLUDE_RENDITIONS",
    MIN_BOTTOM_RENDITION_SIZE: "MIN_BOTTOM_RENDITION_SIZE",
    MIN_TOP_RENDITION_SIZE: "MIN_TOP_RENDITION_SIZE",
};
const BurninSubtitleAlignment = {
    AUTO: "AUTO",
    CENTERED: "CENTERED",
    LEFT: "LEFT",
};
const BurninSubtitleApplyFontColor = {
    ALL_TEXT: "ALL_TEXT",
    WHITE_TEXT_ONLY: "WHITE_TEXT_ONLY",
};
const BurninSubtitleBackgroundColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
const BurninSubtitleFallbackFont = {
    BEST_MATCH: "BEST_MATCH",
    MONOSPACED_SANSSERIF: "MONOSPACED_SANSSERIF",
    MONOSPACED_SERIF: "MONOSPACED_SERIF",
    PROPORTIONAL_SANSSERIF: "PROPORTIONAL_SANSSERIF",
    PROPORTIONAL_SERIF: "PROPORTIONAL_SERIF",
};
const BurninSubtitleFontColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    HEX: "HEX",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
const FontScript = {
    AUTOMATIC: "AUTOMATIC",
    HANS: "HANS",
    HANT: "HANT",
};
const BurninSubtitleOutlineColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
const RemoveRubyReserveAttributes = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const BurninSubtitleShadowColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
const BurnInSubtitleStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const BurninSubtitleTeletextSpacing = {
    AUTO: "AUTO",
    FIXED_GRID: "FIXED_GRID",
    PROPORTIONAL: "PROPORTIONAL",
};
const CaptionDestinationType = {
    BURN_IN: "BURN_IN",
    DVB_SUB: "DVB_SUB",
    EMBEDDED: "EMBEDDED",
    EMBEDDED_PLUS_SCTE20: "EMBEDDED_PLUS_SCTE20",
    IMSC: "IMSC",
    SCC: "SCC",
    SCTE20_PLUS_EMBEDDED: "SCTE20_PLUS_EMBEDDED",
    SMI: "SMI",
    SRT: "SRT",
    TELETEXT: "TELETEXT",
    TTML: "TTML",
    WEBVTT: "WEBVTT",
};
const DvbSubtitleAlignment = {
    AUTO: "AUTO",
    CENTERED: "CENTERED",
    LEFT: "LEFT",
};
const DvbSubtitleApplyFontColor = {
    ALL_TEXT: "ALL_TEXT",
    WHITE_TEXT_ONLY: "WHITE_TEXT_ONLY",
};
const DvbSubtitleBackgroundColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
const DvbddsHandling = {
    NONE: "NONE",
    NO_DISPLAY_WINDOW: "NO_DISPLAY_WINDOW",
    SPECIFIED: "SPECIFIED",
    SPECIFIED_OPTIMAL: "SPECIFIED_OPTIMAL",
};
const DvbSubSubtitleFallbackFont = {
    BEST_MATCH: "BEST_MATCH",
    MONOSPACED_SANSSERIF: "MONOSPACED_SANSSERIF",
    MONOSPACED_SERIF: "MONOSPACED_SERIF",
    PROPORTIONAL_SANSSERIF: "PROPORTIONAL_SANSSERIF",
    PROPORTIONAL_SERIF: "PROPORTIONAL_SERIF",
};
const DvbSubtitleFontColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    HEX: "HEX",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
const DvbSubtitleOutlineColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    BLUE: "BLUE",
    GREEN: "GREEN",
    RED: "RED",
    WHITE: "WHITE",
    YELLOW: "YELLOW",
};
const DvbSubtitleShadowColor = {
    AUTO: "AUTO",
    BLACK: "BLACK",
    NONE: "NONE",
    WHITE: "WHITE",
};
const DvbSubtitleStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DvbSubtitlingType = {
    HEARING_IMPAIRED: "HEARING_IMPAIRED",
    STANDARD: "STANDARD",
};
const DvbSubtitleTeletextSpacing = {
    AUTO: "AUTO",
    FIXED_GRID: "FIXED_GRID",
    PROPORTIONAL: "PROPORTIONAL",
};
const ImscAccessibilitySubs = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ImscStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const SccDestinationFramerate = {
    FRAMERATE_23_97: "FRAMERATE_23_97",
    FRAMERATE_24: "FRAMERATE_24",
    FRAMERATE_25: "FRAMERATE_25",
    FRAMERATE_29_97_DROPFRAME: "FRAMERATE_29_97_DROPFRAME",
    FRAMERATE_29_97_NON_DROPFRAME: "FRAMERATE_29_97_NON_DROPFRAME",
};
const SrtStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const TeletextPageType = {
    PAGE_TYPE_ADDL_INFO: "PAGE_TYPE_ADDL_INFO",
    PAGE_TYPE_HEARING_IMPAIRED_SUBTITLE: "PAGE_TYPE_HEARING_IMPAIRED_SUBTITLE",
    PAGE_TYPE_INITIAL: "PAGE_TYPE_INITIAL",
    PAGE_TYPE_PROGRAM_SCHEDULE: "PAGE_TYPE_PROGRAM_SCHEDULE",
    PAGE_TYPE_SUBTITLE: "PAGE_TYPE_SUBTITLE",
};
const TtmlStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const WebvttAccessibilitySubs = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const WebvttStylePassthrough = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    MERGE: "MERGE",
    STRICT: "STRICT",
};
const ColorSpace = {
    FOLLOW: "FOLLOW",
    HDR10: "HDR10",
    HLG_2020: "HLG_2020",
    P3D65_HDR: "P3D65_HDR",
    P3D65_SDR: "P3D65_SDR",
    P3DCI: "P3DCI",
    REC_601: "REC_601",
    REC_709: "REC_709",
};
const FrameMetricType = {
    MS_SSIM: "MS_SSIM",
    PSNR: "PSNR",
    PSNR_HVS: "PSNR_HVS",
    QVBR: "QVBR",
    SHOT_CHANGE: "SHOT_CHANGE",
    SSIM: "SSIM",
    VMAF: "VMAF",
};
const HlsAdMarkers = {
    ELEMENTAL: "ELEMENTAL",
    ELEMENTAL_SCTE35: "ELEMENTAL_SCTE35",
};
const AdvancedInputFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AdvancedInputFilterAddTexture = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AdvancedInputFilterSharpen = {
    HIGH: "HIGH",
    LOW: "LOW",
    OFF: "OFF",
};
const AudioDurationCorrection = {
    AUTO: "AUTO",
    DISABLED: "DISABLED",
    FORCE: "FORCE",
    FRAME: "FRAME",
    TRACK: "TRACK",
};
const AudioDefaultSelection = {
    DEFAULT: "DEFAULT",
    NOT_DEFAULT: "NOT_DEFAULT",
};
const AudioSelectorType = {
    ALL_PCM: "ALL_PCM",
    HLS_RENDITION_GROUP: "HLS_RENDITION_GROUP",
    LANGUAGE_CODE: "LANGUAGE_CODE",
    PID: "PID",
    STREAM: "STREAM",
    TRACK: "TRACK",
};
const AncillaryConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
const AncillaryTerminateCaptions = {
    DISABLED: "DISABLED",
    END_OF_INPUT: "END_OF_INPUT",
};
const EmbeddedConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
const EmbeddedTerminateCaptions = {
    DISABLED: "DISABLED",
    END_OF_INPUT: "END_OF_INPUT",
};
const CaptionSourceByteRateLimit = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const FileSourceConvert608To708 = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
const CaptionSourceConvertPaintOnToPopOn = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const FileSourceTimeDeltaUnits = {
    MILLISECONDS: "MILLISECONDS",
    SECONDS: "SECONDS",
};
const CaptionSourceUpconvertSTLToTeletext = {
    DISABLED: "DISABLED",
    UPCONVERT: "UPCONVERT",
};
const CaptionSourceType = {
    ANCILLARY: "ANCILLARY",
    DVB_SUB: "DVB_SUB",
    EMBEDDED: "EMBEDDED",
    IMSC: "IMSC",
    NULL_SOURCE: "NULL_SOURCE",
    SCC: "SCC",
    SCTE20: "SCTE20",
    SMI: "SMI",
    SMPTE_TT: "SMPTE_TT",
    SRT: "SRT",
    STL: "STL",
    TELETEXT: "TELETEXT",
    TTML: "TTML",
    TT_3GPP: "TT_3GPP",
    WEBVTT: "WEBVTT",
};
const InputDeblockFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DecryptionMode = {
    AES_CBC: "AES_CBC",
    AES_CTR: "AES_CTR",
    AES_GCM: "AES_GCM",
};
const InputDenoiseFilter = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DynamicAudioSelectorType = {
    ALL_TRACKS: "ALL_TRACKS",
    LANGUAGE_CODE: "LANGUAGE_CODE",
};
const InputFilterEnable = {
    AUTO: "AUTO",
    DISABLE: "DISABLE",
    FORCE: "FORCE",
};
const InputScanType = {
    AUTO: "AUTO",
    PSF: "PSF",
};
const InputPsiControl = {
    IGNORE_PSI: "IGNORE_PSI",
    USE_PSI: "USE_PSI",
};
const TamsGapHandling = {
    FILL_WITH_BLACK: "FILL_WITH_BLACK",
    HOLD_LAST_FRAME: "HOLD_LAST_FRAME",
    SKIP_GAPS: "SKIP_GAPS",
};
const InputTimecodeSource = {
    EMBEDDED: "EMBEDDED",
    SPECIFIEDSTART: "SPECIFIEDSTART",
    ZEROBASED: "ZEROBASED",
};
const VideoOverlayUnit = {
    PERCENTAGE: "PERCENTAGE",
    PIXELS: "PIXELS",
};
const VideoOverlayPlayBackMode = {
    ONCE: "ONCE",
    REPEAT: "REPEAT",
};
const AlphaBehavior = {
    DISCARD: "DISCARD",
    REMAP_TO_LUMA: "REMAP_TO_LUMA",
};
const ColorSpaceUsage = {
    FALLBACK: "FALLBACK",
    FORCE: "FORCE",
};
const EmbeddedTimecodeOverride = {
    NONE: "NONE",
    USE_MDPM: "USE_MDPM",
};
const PadVideo = {
    BLACK: "BLACK",
    DISABLED: "DISABLED",
};
const InputRotate = {
    AUTO: "AUTO",
    DEGREES_180: "DEGREES_180",
    DEGREES_270: "DEGREES_270",
    DEGREES_90: "DEGREES_90",
    DEGREE_0: "DEGREE_0",
};
const InputSampleRange = {
    FOLLOW: "FOLLOW",
    FULL_RANGE: "FULL_RANGE",
    LIMITED_RANGE: "LIMITED_RANGE",
};
const VideoSelectorType = {
    AUTO: "AUTO",
    STREAM: "STREAM",
};
const AccelerationMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    PREFERRED: "PREFERRED",
};
const AccelerationStatus = {
    ACCELERATED: "ACCELERATED",
    IN_PROGRESS: "IN_PROGRESS",
    NOT_ACCELERATED: "NOT_ACCELERATED",
    NOT_APPLICABLE: "NOT_APPLICABLE",
};
const BillingTagsSource = {
    JOB: "JOB",
    JOB_TEMPLATE: "JOB_TEMPLATE",
    PRESET: "PRESET",
    QUEUE: "QUEUE",
};
const JobPhase = {
    PROBING: "PROBING",
    TRANSCODING: "TRANSCODING",
    UPLOADING: "UPLOADING",
};
const CopyProtectionAction = {
    PASSTHROUGH: "PASSTHROUGH",
    STRIP: "STRIP",
};
const VchipAction = {
    PASSTHROUGH: "PASSTHROUGH",
    STRIP: "STRIP",
};
const MotionImageInsertionMode = {
    MOV: "MOV",
    PNG: "PNG",
};
const MotionImagePlayback = {
    ONCE: "ONCE",
    REPEAT: "REPEAT",
};
const NielsenActiveWatermarkProcessType = {
    CBET: "CBET",
    NAES2_AND_NW: "NAES2_AND_NW",
    NAES2_AND_NW_AND_CBET: "NAES2_AND_NW_AND_CBET",
};
const NielsenSourceWatermarkStatusType = {
    CLEAN: "CLEAN",
    WATERMARKED: "WATERMARKED",
};
const NielsenUniqueTicPerAudioTrackType = {
    RESERVE_UNIQUE_TICS_PER_TRACK: "RESERVE_UNIQUE_TICS_PER_TRACK",
    SAME_TICS_PER_TRACK: "SAME_TICS_PER_TRACK",
};
const CmafClientCache = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CmafCodecSpecification = {
    RFC_4281: "RFC_4281",
    RFC_6381: "RFC_6381",
};
const DashManifestStyle = {
    BASIC: "BASIC",
    COMPACT: "COMPACT",
    DISTINCT: "DISTINCT",
    FULL: "FULL",
};
const S3ObjectCannedAcl = {
    AUTHENTICATED_READ: "AUTHENTICATED_READ",
    BUCKET_OWNER_FULL_CONTROL: "BUCKET_OWNER_FULL_CONTROL",
    BUCKET_OWNER_READ: "BUCKET_OWNER_READ",
    PUBLIC_READ: "PUBLIC_READ",
};
const S3ServerSideEncryptionType = {
    SERVER_SIDE_ENCRYPTION_KMS: "SERVER_SIDE_ENCRYPTION_KMS",
    SERVER_SIDE_ENCRYPTION_S3: "SERVER_SIDE_ENCRYPTION_S3",
};
const S3StorageClass = {
    DEEP_ARCHIVE: "DEEP_ARCHIVE",
    GLACIER: "GLACIER",
    INTELLIGENT_TIERING: "INTELLIGENT_TIERING",
    ONEZONE_IA: "ONEZONE_IA",
    REDUCED_REDUNDANCY: "REDUCED_REDUNDANCY",
    STANDARD: "STANDARD",
    STANDARD_IA: "STANDARD_IA",
};
const CmafEncryptionType = {
    AES_CTR: "AES_CTR",
    SAMPLE_AES: "SAMPLE_AES",
};
const CmafInitializationVectorInManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const PresetSpeke20Audio = {
    PRESET_AUDIO_1: "PRESET_AUDIO_1",
    PRESET_AUDIO_2: "PRESET_AUDIO_2",
    PRESET_AUDIO_3: "PRESET_AUDIO_3",
    SHARED: "SHARED",
    UNENCRYPTED: "UNENCRYPTED",
};
const PresetSpeke20Video = {
    PRESET_VIDEO_1: "PRESET_VIDEO_1",
    PRESET_VIDEO_2: "PRESET_VIDEO_2",
    PRESET_VIDEO_3: "PRESET_VIDEO_3",
    PRESET_VIDEO_4: "PRESET_VIDEO_4",
    PRESET_VIDEO_5: "PRESET_VIDEO_5",
    PRESET_VIDEO_6: "PRESET_VIDEO_6",
    PRESET_VIDEO_7: "PRESET_VIDEO_7",
    PRESET_VIDEO_8: "PRESET_VIDEO_8",
    SHARED: "SHARED",
    UNENCRYPTED: "UNENCRYPTED",
};
const CmafKeyProviderType = {
    SPEKE: "SPEKE",
    STATIC_KEY: "STATIC_KEY",
};
const CmafImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
const CmafIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
const CmafManifestCompression = {
    GZIP: "GZIP",
    NONE: "NONE",
};
const CmafManifestDurationFormat = {
    FLOATING_POINT: "FLOATING_POINT",
    INTEGER: "INTEGER",
};
const CmafMpdManifestBandwidthType = {
    AVERAGE: "AVERAGE",
    MAX: "MAX",
};
const CmafMpdProfile = {
    MAIN_PROFILE: "MAIN_PROFILE",
    ON_DEMAND_PROFILE: "ON_DEMAND_PROFILE",
};
const CmafPtsOffsetHandlingForBFrames = {
    MATCH_INITIAL_PTS: "MATCH_INITIAL_PTS",
    ZERO_BASED: "ZERO_BASED",
};
const CmafSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
const CmafSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
    MATCH: "MATCH",
};
const CmafStreamInfResolution = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const CmafTargetDurationCompatibilityMode = {
    LEGACY: "LEGACY",
    SPEC_COMPLIANT: "SPEC_COMPLIANT",
};
const CmafVideoCompositionOffsets = {
    SIGNED: "SIGNED",
    UNSIGNED: "UNSIGNED",
};
const CmafWriteDASHManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CmafWriteHLSManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CmafWriteSegmentTimelineInRepresentation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DashIsoGroupAudioChannelConfigSchemeIdUri = {
    DOLBY_CHANNEL_CONFIGURATION: "DOLBY_CHANNEL_CONFIGURATION",
    MPEG_CHANNEL_CONFIGURATION: "MPEG_CHANNEL_CONFIGURATION",
};
const DashIsoPlaybackDeviceCompatibility = {
    CENC_V1: "CENC_V1",
    UNENCRYPTED_SEI: "UNENCRYPTED_SEI",
};
const DashIsoHbbtvCompliance = {
    HBBTV_1_5: "HBBTV_1_5",
    NONE: "NONE",
};
const DashIsoImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
const DashIsoIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
const DashIsoMpdManifestBandwidthType = {
    AVERAGE: "AVERAGE",
    MAX: "MAX",
};
const DashIsoMpdProfile = {
    MAIN_PROFILE: "MAIN_PROFILE",
    ON_DEMAND_PROFILE: "ON_DEMAND_PROFILE",
};
const DashIsoPtsOffsetHandlingForBFrames = {
    MATCH_INITIAL_PTS: "MATCH_INITIAL_PTS",
    ZERO_BASED: "ZERO_BASED",
};
const DashIsoSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
const DashIsoSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
    MATCH: "MATCH",
};
const DashIsoVideoCompositionOffsets = {
    SIGNED: "SIGNED",
    UNSIGNED: "UNSIGNED",
};
const DashIsoWriteSegmentTimelineInRepresentation = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const HlsAudioOnlyHeader = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const HlsCaptionLanguageSetting = {
    INSERT: "INSERT",
    NONE: "NONE",
    OMIT: "OMIT",
};
const HlsCaptionSegmentLengthControl = {
    LARGE_SEGMENTS: "LARGE_SEGMENTS",
    MATCH_VIDEO: "MATCH_VIDEO",
};
const HlsClientCache = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const HlsCodecSpecification = {
    RFC_4281: "RFC_4281",
    RFC_6381: "RFC_6381",
};
const HlsDirectoryStructure = {
    SINGLE_DIRECTORY: "SINGLE_DIRECTORY",
    SUBDIRECTORY_PER_STREAM: "SUBDIRECTORY_PER_STREAM",
};
const HlsEncryptionType = {
    AES128: "AES128",
    SAMPLE_AES: "SAMPLE_AES",
};
const HlsInitializationVectorInManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const HlsOfflineEncrypted = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const HlsKeyProviderType = {
    SPEKE: "SPEKE",
    STATIC_KEY: "STATIC_KEY",
};
const HlsImageBasedTrickPlay = {
    ADVANCED: "ADVANCED",
    NONE: "NONE",
    THUMBNAIL: "THUMBNAIL",
    THUMBNAIL_AND_FULLFRAME: "THUMBNAIL_AND_FULLFRAME",
};
const HlsIntervalCadence = {
    FOLLOW_CUSTOM: "FOLLOW_CUSTOM",
    FOLLOW_IFRAME: "FOLLOW_IFRAME",
};
const HlsManifestCompression = {
    GZIP: "GZIP",
    NONE: "NONE",
};
const HlsManifestDurationFormat = {
    FLOATING_POINT: "FLOATING_POINT",
    INTEGER: "INTEGER",
};
const HlsOutputSelection = {
    MANIFESTS_AND_SEGMENTS: "MANIFESTS_AND_SEGMENTS",
    SEGMENTS_ONLY: "SEGMENTS_ONLY",
};
const HlsProgramDateTime = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const HlsProgressiveWriteHlsManifest = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const HlsSegmentControl = {
    SEGMENTED_FILES: "SEGMENTED_FILES",
    SINGLE_FILE: "SINGLE_FILE",
};
const HlsSegmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
    MATCH: "MATCH",
};
const HlsStreamInfResolution = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const HlsTargetDurationCompatibilityMode = {
    LEGACY: "LEGACY",
    SPEC_COMPLIANT: "SPEC_COMPLIANT",
};
const HlsTimedMetadataId3Frame = {
    NONE: "NONE",
    PRIV: "PRIV",
    TDRL: "TDRL",
};
const MsSmoothAudioDeduplication = {
    COMBINE_DUPLICATE_STREAMS: "COMBINE_DUPLICATE_STREAMS",
    NONE: "NONE",
};
const MsSmoothFragmentLengthControl = {
    EXACT: "EXACT",
    GOP_MULTIPLE: "GOP_MULTIPLE",
};
const MsSmoothManifestEncoding = {
    UTF16: "UTF16",
    UTF8: "UTF8",
};
const OutputGroupType = {
    CMAF_GROUP_SETTINGS: "CMAF_GROUP_SETTINGS",
    DASH_ISO_GROUP_SETTINGS: "DASH_ISO_GROUP_SETTINGS",
    FILE_GROUP_SETTINGS: "FILE_GROUP_SETTINGS",
    HLS_GROUP_SETTINGS: "HLS_GROUP_SETTINGS",
    MS_SMOOTH_GROUP_SETTINGS: "MS_SMOOTH_GROUP_SETTINGS",
};
const CmfcAudioDuration = {
    DEFAULT_CODEC_DURATION: "DEFAULT_CODEC_DURATION",
    MATCH_VIDEO_DURATION: "MATCH_VIDEO_DURATION",
};
const CmfcAudioTrackType = {
    ALTERNATE_AUDIO_AUTO_SELECT: "ALTERNATE_AUDIO_AUTO_SELECT",
    ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT: "ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT",
    ALTERNATE_AUDIO_NOT_AUTO_SELECT: "ALTERNATE_AUDIO_NOT_AUTO_SELECT",
    AUDIO_ONLY_VARIANT_STREAM: "AUDIO_ONLY_VARIANT_STREAM",
};
const CmfcC2paManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const CmfcDescriptiveVideoServiceFlag = {
    DONT_FLAG: "DONT_FLAG",
    FLAG: "FLAG",
};
const CmfcIFrameOnlyManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const CmfcKlvMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const CmfcManifestMetadataSignaling = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const CmfcScte35Esam = {
    INSERT: "INSERT",
    NONE: "NONE",
};
const CmfcScte35Source = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const CmfcTimedMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const CmfcTimedMetadataBoxVersion = {
    VERSION_0: "VERSION_0",
    VERSION_1: "VERSION_1",
};
const ContainerType = {
    CMFC: "CMFC",
    F4V: "F4V",
    GIF: "GIF",
    ISMV: "ISMV",
    M2TS: "M2TS",
    M3U8: "M3U8",
    MOV: "MOV",
    MP4: "MP4",
    MPD: "MPD",
    MXF: "MXF",
    OGG: "OGG",
    RAW: "RAW",
    WEBM: "WEBM",
    Y4M: "Y4M",
};
const F4vMoovPlacement = {
    NORMAL: "NORMAL",
    PROGRESSIVE_DOWNLOAD: "PROGRESSIVE_DOWNLOAD",
};
const M2tsAudioBufferModel = {
    ATSC: "ATSC",
    DVB: "DVB",
};
const M2tsAudioDuration = {
    DEFAULT_CODEC_DURATION: "DEFAULT_CODEC_DURATION",
    MATCH_VIDEO_DURATION: "MATCH_VIDEO_DURATION",
};
const M2tsBufferModel = {
    MULTIPLEX: "MULTIPLEX",
    NONE: "NONE",
};
const M2tsDataPtsControl = {
    ALIGN_TO_VIDEO: "ALIGN_TO_VIDEO",
    AUTO: "AUTO",
};
const OutputSdt = {
    SDT_FOLLOW: "SDT_FOLLOW",
    SDT_FOLLOW_IF_PRESENT: "SDT_FOLLOW_IF_PRESENT",
    SDT_MANUAL: "SDT_MANUAL",
    SDT_NONE: "SDT_NONE",
};
const M2tsEbpAudioInterval = {
    VIDEO_AND_FIXED_INTERVALS: "VIDEO_AND_FIXED_INTERVALS",
    VIDEO_INTERVAL: "VIDEO_INTERVAL",
};
const M2tsEbpPlacement = {
    VIDEO_AND_AUDIO_PIDS: "VIDEO_AND_AUDIO_PIDS",
    VIDEO_PID: "VIDEO_PID",
};
const M2tsEsRateInPes = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const M2tsForceTsVideoEbpOrder = {
    DEFAULT: "DEFAULT",
    FORCE: "FORCE",
};
const M2tsKlvMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const M2tsNielsenId3 = {
    INSERT: "INSERT",
    NONE: "NONE",
};
const M2tsPcrControl = {
    CONFIGURED_PCR_PERIOD: "CONFIGURED_PCR_PERIOD",
    PCR_EVERY_PES_PACKET: "PCR_EVERY_PES_PACKET",
};
const M2tsPreventBufferUnderflow = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const TsPtsOffset = {
    AUTO: "AUTO",
    MILLISECONDS: "MILLISECONDS",
    SECONDS: "SECONDS",
};
const M2tsRateMode = {
    CBR: "CBR",
    VBR: "VBR",
};
const M2tsScte35Source = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const M2tsSegmentationMarkers = {
    EBP: "EBP",
    EBP_LEGACY: "EBP_LEGACY",
    NONE: "NONE",
    PSI_SEGSTART: "PSI_SEGSTART",
    RAI_ADAPT: "RAI_ADAPT",
    RAI_SEGSTART: "RAI_SEGSTART",
};
const M2tsSegmentationStyle = {
    MAINTAIN_CADENCE: "MAINTAIN_CADENCE",
    RESET_CADENCE: "RESET_CADENCE",
};
const M3u8AudioDuration = {
    DEFAULT_CODEC_DURATION: "DEFAULT_CODEC_DURATION",
    MATCH_VIDEO_DURATION: "MATCH_VIDEO_DURATION",
};
const M3u8DataPtsControl = {
    ALIGN_TO_VIDEO: "ALIGN_TO_VIDEO",
    AUTO: "AUTO",
};
const M3u8NielsenId3 = {
    INSERT: "INSERT",
    NONE: "NONE",
};
const M3u8PcrControl = {
    CONFIGURED_PCR_PERIOD: "CONFIGURED_PCR_PERIOD",
    PCR_EVERY_PES_PACKET: "PCR_EVERY_PES_PACKET",
};
const M3u8Scte35Source = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const TimedMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const MovClapAtom = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const MovCslgAtom = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const MovMpeg2FourCCControl = {
    MPEG: "MPEG",
    XDCAM: "XDCAM",
};
const MovPaddingControl = {
    NONE: "NONE",
    OMNEON: "OMNEON",
};
const MovReference = {
    EXTERNAL: "EXTERNAL",
    SELF_CONTAINED: "SELF_CONTAINED",
};
const Mp4C2paManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const Mp4CslgAtom = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const Mp4FreeSpaceBox = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const Mp4MoovPlacement = {
    NORMAL: "NORMAL",
    PROGRESSIVE_DOWNLOAD: "PROGRESSIVE_DOWNLOAD",
};
const MpdAccessibilityCaptionHints = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const MpdAudioDuration = {
    DEFAULT_CODEC_DURATION: "DEFAULT_CODEC_DURATION",
    MATCH_VIDEO_DURATION: "MATCH_VIDEO_DURATION",
};
const MpdC2paManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
};
const MpdCaptionContainerType = {
    FRAGMENTED_MP4: "FRAGMENTED_MP4",
    RAW: "RAW",
};
const MpdKlvMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const MpdManifestMetadataSignaling = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const MpdScte35Esam = {
    INSERT: "INSERT",
    NONE: "NONE",
};
const MpdScte35Source = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const MpdTimedMetadata = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
};
const MpdTimedMetadataBoxVersion = {
    VERSION_0: "VERSION_0",
    VERSION_1: "VERSION_1",
};
const MxfAfdSignaling = {
    COPY_FROM_VIDEO: "COPY_FROM_VIDEO",
    NO_COPY: "NO_COPY",
};
const MxfProfile = {
    D_10: "D_10",
    OP1A: "OP1A",
    XAVC: "XAVC",
    XDCAM: "XDCAM",
    XDCAM_RDD9: "XDCAM_RDD9",
};
const MxfUncompressedAudioWrapping = {
    AES3: "AES3",
    AUTO: "AUTO",
};
const MxfXavcDurationMode = {
    ALLOW_ANY_DURATION: "ALLOW_ANY_DURATION",
    DROP_FRAMES_FOR_COMPLIANCE: "DROP_FRAMES_FOR_COMPLIANCE",
};
const HlsAudioOnlyContainer = {
    AUTOMATIC: "AUTOMATIC",
    M2TS: "M2TS",
};
const HlsAudioTrackType = {
    ALTERNATE_AUDIO_AUTO_SELECT: "ALTERNATE_AUDIO_AUTO_SELECT",
    ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT: "ALTERNATE_AUDIO_AUTO_SELECT_DEFAULT",
    ALTERNATE_AUDIO_NOT_AUTO_SELECT: "ALTERNATE_AUDIO_NOT_AUTO_SELECT",
    AUDIO_ONLY_VARIANT_STREAM: "AUDIO_ONLY_VARIANT_STREAM",
};
const HlsDescriptiveVideoServiceFlag = {
    DONT_FLAG: "DONT_FLAG",
    FLAG: "FLAG",
};
const HlsIFrameOnlyManifest = {
    EXCLUDE: "EXCLUDE",
    INCLUDE: "INCLUDE",
    INCLUDE_AS_TS: "INCLUDE_AS_TS",
};
const AfdSignaling = {
    AUTO: "AUTO",
    FIXED: "FIXED",
    NONE: "NONE",
};
const AntiAlias = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ChromaPositionMode = {
    AUTO: "AUTO",
    FORCE_CENTER: "FORCE_CENTER",
    FORCE_TOP_LEFT: "FORCE_TOP_LEFT",
};
const Av1AdaptiveQuantization = {
    HIGH: "HIGH",
    HIGHER: "HIGHER",
    LOW: "LOW",
    MAX: "MAX",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const Av1BitDepth = {
    BIT_10: "BIT_10",
    BIT_8: "BIT_8",
};
const Av1FilmGrainSynthesis = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Av1FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Av1FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const Av1RateControlMode = {
    QVBR: "QVBR",
};
const Av1SpatialAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AvcIntraClass = {
    CLASS_100: "CLASS_100",
    CLASS_200: "CLASS_200",
    CLASS_4K_2K: "CLASS_4K_2K",
    CLASS_50: "CLASS_50",
};
const AvcIntraUhdQualityTuningLevel = {
    MULTI_PASS: "MULTI_PASS",
    SINGLE_PASS: "SINGLE_PASS",
};
const AvcIntraFramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const AvcIntraFramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const AvcIntraInterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const AvcIntraScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const AvcIntraSlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AvcIntraTelecine = {
    HARD: "HARD",
    NONE: "NONE",
};
const VideoCodec = {
    AV1: "AV1",
    AVC_INTRA: "AVC_INTRA",
    FRAME_CAPTURE: "FRAME_CAPTURE",
    GIF: "GIF",
    H_264: "H_264",
    H_265: "H_265",
    MPEG2: "MPEG2",
    PASSTHROUGH: "PASSTHROUGH",
    PRORES: "PRORES",
    UNCOMPRESSED: "UNCOMPRESSED",
    VC3: "VC3",
    VP8: "VP8",
    VP9: "VP9",
    XAVC: "XAVC",
};
const GifFramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const GifFramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    INTERPOLATE: "INTERPOLATE",
};
const H264AdaptiveQuantization = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    HIGHER: "HIGHER",
    LOW: "LOW",
    MAX: "MAX",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const BandwidthReductionFilterSharpening = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const BandwidthReductionFilterStrength = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const H264CodecLevel = {
    AUTO: "AUTO",
    LEVEL_1: "LEVEL_1",
    LEVEL_1_1: "LEVEL_1_1",
    LEVEL_1_2: "LEVEL_1_2",
    LEVEL_1_3: "LEVEL_1_3",
    LEVEL_2: "LEVEL_2",
    LEVEL_2_1: "LEVEL_2_1",
    LEVEL_2_2: "LEVEL_2_2",
    LEVEL_3: "LEVEL_3",
    LEVEL_3_1: "LEVEL_3_1",
    LEVEL_3_2: "LEVEL_3_2",
    LEVEL_4: "LEVEL_4",
    LEVEL_4_1: "LEVEL_4_1",
    LEVEL_4_2: "LEVEL_4_2",
    LEVEL_5: "LEVEL_5",
    LEVEL_5_1: "LEVEL_5_1",
    LEVEL_5_2: "LEVEL_5_2",
};
const H264CodecProfile = {
    BASELINE: "BASELINE",
    HIGH: "HIGH",
    HIGH_10BIT: "HIGH_10BIT",
    HIGH_422: "HIGH_422",
    HIGH_422_10BIT: "HIGH_422_10BIT",
    MAIN: "MAIN",
};
const H264DynamicSubGop = {
    ADAPTIVE: "ADAPTIVE",
    STATIC: "STATIC",
};
const H264EndOfStreamMarkers = {
    INCLUDE: "INCLUDE",
    SUPPRESS: "SUPPRESS",
};
const H264EntropyEncoding = {
    CABAC: "CABAC",
    CAVLC: "CAVLC",
};
const H264FieldEncoding = {
    FORCE_FIELD: "FORCE_FIELD",
    MBAFF: "MBAFF",
    PAFF: "PAFF",
};
const H264FlickerAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const H264FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const H264GopBReference = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264GopSizeUnits = {
    AUTO: "AUTO",
    FRAMES: "FRAMES",
    SECONDS: "SECONDS",
};
const H264InterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const H264ParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const H264QualityTuningLevel = {
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
    SINGLE_PASS: "SINGLE_PASS",
    SINGLE_PASS_HQ: "SINGLE_PASS_HQ",
};
const H264RateControlMode = {
    CBR: "CBR",
    QVBR: "QVBR",
    VBR: "VBR",
};
const H264RepeatPps = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264SaliencyAwareEncoding = {
    DISABLED: "DISABLED",
    PREFERRED: "PREFERRED",
};
const H264ScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const H264SceneChangeDetect = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    TRANSITION_DETECTION: "TRANSITION_DETECTION",
};
const H264SlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264SpatialAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264Syntax = {
    DEFAULT: "DEFAULT",
    RP2027: "RP2027",
};
const H264Telecine = {
    HARD: "HARD",
    NONE: "NONE",
    SOFT: "SOFT",
};
const H264TemporalAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264UnregisteredSeiTimecode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H264WriteMp4PackagingType = {
    AVC1: "AVC1",
    AVC3: "AVC3",
};
const H265AdaptiveQuantization = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    HIGHER: "HIGHER",
    LOW: "LOW",
    MAX: "MAX",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const H265AlternateTransferFunctionSei = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265CodecLevel = {
    AUTO: "AUTO",
    LEVEL_1: "LEVEL_1",
    LEVEL_2: "LEVEL_2",
    LEVEL_2_1: "LEVEL_2_1",
    LEVEL_3: "LEVEL_3",
    LEVEL_3_1: "LEVEL_3_1",
    LEVEL_4: "LEVEL_4",
    LEVEL_4_1: "LEVEL_4_1",
    LEVEL_5: "LEVEL_5",
    LEVEL_5_1: "LEVEL_5_1",
    LEVEL_5_2: "LEVEL_5_2",
    LEVEL_6: "LEVEL_6",
    LEVEL_6_1: "LEVEL_6_1",
    LEVEL_6_2: "LEVEL_6_2",
};
const H265CodecProfile = {
    MAIN10_HIGH: "MAIN10_HIGH",
    MAIN10_MAIN: "MAIN10_MAIN",
    MAIN_422_10BIT_HIGH: "MAIN_422_10BIT_HIGH",
    MAIN_422_10BIT_MAIN: "MAIN_422_10BIT_MAIN",
    MAIN_422_8BIT_HIGH: "MAIN_422_8BIT_HIGH",
    MAIN_422_8BIT_MAIN: "MAIN_422_8BIT_MAIN",
    MAIN_HIGH: "MAIN_HIGH",
    MAIN_MAIN: "MAIN_MAIN",
};
const H265Deblocking = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265DynamicSubGop = {
    ADAPTIVE: "ADAPTIVE",
    STATIC: "STATIC",
};
const H265EndOfStreamMarkers = {
    INCLUDE: "INCLUDE",
    SUPPRESS: "SUPPRESS",
};
const H265FlickerAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const H265FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const H265GopBReference = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265GopSizeUnits = {
    AUTO: "AUTO",
    FRAMES: "FRAMES",
    SECONDS: "SECONDS",
};
const H265InterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const H265MvOverPictureBoundaries = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265MvTemporalPredictor = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265ParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const H265QualityTuningLevel = {
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
    SINGLE_PASS: "SINGLE_PASS",
    SINGLE_PASS_HQ: "SINGLE_PASS_HQ",
};
const H265RateControlMode = {
    CBR: "CBR",
    QVBR: "QVBR",
    VBR: "VBR",
};
const H265SampleAdaptiveOffsetFilterMode = {
    ADAPTIVE: "ADAPTIVE",
    DEFAULT: "DEFAULT",
    OFF: "OFF",
};
const H265ScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const H265SceneChangeDetect = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    TRANSITION_DETECTION: "TRANSITION_DETECTION",
};
const H265SlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265SpatialAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265Telecine = {
    HARD: "HARD",
    NONE: "NONE",
    SOFT: "SOFT",
};
const H265TemporalAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265TemporalIds = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265TilePadding = {
    NONE: "NONE",
    PADDED: "PADDED",
};
const H265Tiles = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265TreeBlockSize = {
    AUTO: "AUTO",
    TREE_SIZE_32X32: "TREE_SIZE_32X32",
};
const H265UnregisteredSeiTimecode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const H265WriteMp4PackagingType = {
    HEV1: "HEV1",
    HVC1: "HVC1",
};
const Mpeg2AdaptiveQuantization = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const Mpeg2CodecLevel = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    HIGH1440: "HIGH1440",
    LOW: "LOW",
    MAIN: "MAIN",
};
const Mpeg2CodecProfile = {
    MAIN: "MAIN",
    PROFILE_422: "PROFILE_422",
};
const Mpeg2DynamicSubGop = {
    ADAPTIVE: "ADAPTIVE",
    STATIC: "STATIC",
};
const Mpeg2FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Mpeg2FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const Mpeg2GopSizeUnits = {
    FRAMES: "FRAMES",
    SECONDS: "SECONDS",
};
const Mpeg2InterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const Mpeg2IntraDcPrecision = {
    AUTO: "AUTO",
    INTRA_DC_PRECISION_10: "INTRA_DC_PRECISION_10",
    INTRA_DC_PRECISION_11: "INTRA_DC_PRECISION_11",
    INTRA_DC_PRECISION_8: "INTRA_DC_PRECISION_8",
    INTRA_DC_PRECISION_9: "INTRA_DC_PRECISION_9",
};
const Mpeg2ParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Mpeg2QualityTuningLevel = {
    MULTI_PASS: "MULTI_PASS",
    SINGLE_PASS: "SINGLE_PASS",
};
const Mpeg2RateControlMode = {
    CBR: "CBR",
    VBR: "VBR",
};
const Mpeg2ScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const Mpeg2SceneChangeDetect = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Mpeg2SlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Mpeg2SpatialAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Mpeg2Syntax = {
    DEFAULT: "DEFAULT",
    D_10: "D_10",
};
const Mpeg2Telecine = {
    HARD: "HARD",
    NONE: "NONE",
    SOFT: "SOFT",
};
const Mpeg2TemporalAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const FrameControl = {
    NEAREST_IDRFRAME: "NEAREST_IDRFRAME",
    NEAREST_IFRAME: "NEAREST_IFRAME",
};
const VideoSelectorMode = {
    AUTO: "AUTO",
    REMUX_ALL: "REMUX_ALL",
};
const ProresChromaSampling = {
    PRESERVE_444_SAMPLING: "PRESERVE_444_SAMPLING",
    SUBSAMPLE_TO_422: "SUBSAMPLE_TO_422",
};
const ProresCodecProfile = {
    APPLE_PRORES_422: "APPLE_PRORES_422",
    APPLE_PRORES_422_HQ: "APPLE_PRORES_422_HQ",
    APPLE_PRORES_422_LT: "APPLE_PRORES_422_LT",
    APPLE_PRORES_422_PROXY: "APPLE_PRORES_422_PROXY",
    APPLE_PRORES_4444: "APPLE_PRORES_4444",
    APPLE_PRORES_4444_XQ: "APPLE_PRORES_4444_XQ",
};
const ProresFramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const ProresFramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const ProresInterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const ProresParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const ProresScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const ProresSlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ProresTelecine = {
    HARD: "HARD",
    NONE: "NONE",
};
const UncompressedFourcc = {
    I420: "I420",
    I422: "I422",
    I444: "I444",
};
const UncompressedFramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const UncompressedFramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const UncompressedInterlaceMode = {
    INTERLACED: "INTERLACED",
    PROGRESSIVE: "PROGRESSIVE",
};
const UncompressedScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const UncompressedSlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const UncompressedTelecine = {
    HARD: "HARD",
    NONE: "NONE",
};
const Vc3FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Vc3FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const Vc3InterlaceMode = {
    INTERLACED: "INTERLACED",
    PROGRESSIVE: "PROGRESSIVE",
};
const Vc3ScanTypeConversionMode = {
    INTERLACED: "INTERLACED",
    INTERLACED_OPTIMIZE: "INTERLACED_OPTIMIZE",
};
const Vc3SlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Vc3Telecine = {
    HARD: "HARD",
    NONE: "NONE",
};
const Vc3Class = {
    CLASS_145_8BIT: "CLASS_145_8BIT",
    CLASS_220_10BIT: "CLASS_220_10BIT",
    CLASS_220_8BIT: "CLASS_220_8BIT",
};
const Vp8FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Vp8FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const Vp8ParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Vp8QualityTuningLevel = {
    MULTI_PASS: "MULTI_PASS",
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
};
const Vp8RateControlMode = {
    VBR: "VBR",
};
const Vp9FramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Vp9FramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const Vp9ParControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const Vp9QualityTuningLevel = {
    MULTI_PASS: "MULTI_PASS",
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
};
const Vp9RateControlMode = {
    VBR: "VBR",
};
const XavcAdaptiveQuantization = {
    AUTO: "AUTO",
    HIGH: "HIGH",
    HIGHER: "HIGHER",
    LOW: "LOW",
    MAX: "MAX",
    MEDIUM: "MEDIUM",
    OFF: "OFF",
};
const XavcEntropyEncoding = {
    AUTO: "AUTO",
    CABAC: "CABAC",
    CAVLC: "CAVLC",
};
const XavcFramerateControl = {
    INITIALIZE_FROM_SOURCE: "INITIALIZE_FROM_SOURCE",
    SPECIFIED: "SPECIFIED",
};
const XavcFramerateConversionAlgorithm = {
    DUPLICATE_DROP: "DUPLICATE_DROP",
    FRAMEFORMER: "FRAMEFORMER",
    INTERPOLATE: "INTERPOLATE",
    MAINTAIN_FRAME_COUNT: "MAINTAIN_FRAME_COUNT",
};
const XavcProfile = {
    XAVC_4K: "XAVC_4K",
    XAVC_4K_INTRA_CBG: "XAVC_4K_INTRA_CBG",
    XAVC_4K_INTRA_VBR: "XAVC_4K_INTRA_VBR",
    XAVC_HD: "XAVC_HD",
    XAVC_HD_INTRA_CBG: "XAVC_HD_INTRA_CBG",
};
const XavcSlowPal = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const XavcSpatialAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const XavcTemporalAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Xavc4kIntraCbgProfileClass = {
    CLASS_100: "CLASS_100",
    CLASS_300: "CLASS_300",
    CLASS_480: "CLASS_480",
};
const Xavc4kIntraVbrProfileClass = {
    CLASS_100: "CLASS_100",
    CLASS_300: "CLASS_300",
    CLASS_480: "CLASS_480",
};
const Xavc4kProfileBitrateClass = {
    BITRATE_CLASS_100: "BITRATE_CLASS_100",
    BITRATE_CLASS_140: "BITRATE_CLASS_140",
    BITRATE_CLASS_200: "BITRATE_CLASS_200",
};
const Xavc4kProfileCodecProfile = {
    HIGH: "HIGH",
    HIGH_422: "HIGH_422",
};
const XavcFlickerAdaptiveQuantization = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const XavcGopBReference = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const Xavc4kProfileQualityTuningLevel = {
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
    SINGLE_PASS: "SINGLE_PASS",
    SINGLE_PASS_HQ: "SINGLE_PASS_HQ",
};
const XavcHdIntraCbgProfileClass = {
    CLASS_100: "CLASS_100",
    CLASS_200: "CLASS_200",
    CLASS_50: "CLASS_50",
};
const XavcHdProfileBitrateClass = {
    BITRATE_CLASS_25: "BITRATE_CLASS_25",
    BITRATE_CLASS_35: "BITRATE_CLASS_35",
    BITRATE_CLASS_50: "BITRATE_CLASS_50",
};
const XavcInterlaceMode = {
    BOTTOM_FIELD: "BOTTOM_FIELD",
    FOLLOW_BOTTOM_FIELD: "FOLLOW_BOTTOM_FIELD",
    FOLLOW_TOP_FIELD: "FOLLOW_TOP_FIELD",
    PROGRESSIVE: "PROGRESSIVE",
    TOP_FIELD: "TOP_FIELD",
};
const XavcHdProfileQualityTuningLevel = {
    MULTI_PASS_HQ: "MULTI_PASS_HQ",
    SINGLE_PASS: "SINGLE_PASS",
    SINGLE_PASS_HQ: "SINGLE_PASS_HQ",
};
const XavcHdProfileTelecine = {
    HARD: "HARD",
    NONE: "NONE",
};
const ColorMetadata = {
    IGNORE: "IGNORE",
    INSERT: "INSERT",
};
const DropFrameTimecode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const RespondToAfd = {
    NONE: "NONE",
    PASSTHROUGH: "PASSTHROUGH",
    RESPOND: "RESPOND",
};
const ScalingBehavior = {
    DEFAULT: "DEFAULT",
    FILL: "FILL",
    FIT: "FIT",
    FIT_NO_UPSCALE: "FIT_NO_UPSCALE",
    STRETCH_TO_OUTPUT: "STRETCH_TO_OUTPUT",
};
const VideoTimecodeInsertion = {
    DISABLED: "DISABLED",
    PIC_TIMING_SEI: "PIC_TIMING_SEI",
};
const TimecodeTrack = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ColorSpaceConversion = {
    FORCE_601: "FORCE_601",
    FORCE_709: "FORCE_709",
    FORCE_HDR10: "FORCE_HDR10",
    FORCE_HLG_2020: "FORCE_HLG_2020",
    FORCE_P3D65_HDR: "FORCE_P3D65_HDR",
    FORCE_P3D65_SDR: "FORCE_P3D65_SDR",
    FORCE_P3DCI: "FORCE_P3DCI",
    NONE: "NONE",
};
const HDRToSDRToneMapper = {
    PRESERVE_DETAILS: "PRESERVE_DETAILS",
    VIBRANT: "VIBRANT",
};
const SampleRangeConversion = {
    LIMITED_RANGE_CLIP: "LIMITED_RANGE_CLIP",
    LIMITED_RANGE_SQUEEZE: "LIMITED_RANGE_SQUEEZE",
    NONE: "NONE",
};
const DeinterlaceAlgorithm = {
    BLEND: "BLEND",
    BLEND_TICKER: "BLEND_TICKER",
    INTERPOLATE: "INTERPOLATE",
    INTERPOLATE_TICKER: "INTERPOLATE_TICKER",
    LINEAR_INTERPOLATION: "LINEAR_INTERPOLATION",
};
const DeinterlacerControl = {
    FORCE_ALL_FRAMES: "FORCE_ALL_FRAMES",
    NORMAL: "NORMAL",
};
const DeinterlacerMode = {
    ADAPTIVE: "ADAPTIVE",
    DEINTERLACE: "DEINTERLACE",
    INVERSE_TELECINE: "INVERSE_TELECINE",
};
const DolbyVisionCompatibility = {
    DUPLICATE_STREAM: "DUPLICATE_STREAM",
    SUPPLEMENTAL_CODECS: "SUPPLEMENTAL_CODECS",
};
const DolbyVisionLevel6Mode = {
    PASSTHROUGH: "PASSTHROUGH",
    RECALCULATE: "RECALCULATE",
    SPECIFY: "SPECIFY",
};
const DolbyVisionMapping = {
    HDR10_1000: "HDR10_1000",
    HDR10_NOMAP: "HDR10_NOMAP",
};
const DolbyVisionProfile = {
    PROFILE_5: "PROFILE_5",
    PROFILE_8_1: "PROFILE_8_1",
};
const NoiseReducerFilter = {
    BILATERAL: "BILATERAL",
    CONSERVE: "CONSERVE",
    GAUSSIAN: "GAUSSIAN",
    LANCZOS: "LANCZOS",
    MEAN: "MEAN",
    SHARPEN: "SHARPEN",
    SPATIAL: "SPATIAL",
    TEMPORAL: "TEMPORAL",
};
const NoiseFilterPostTemporalSharpening = {
    AUTO: "AUTO",
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const NoiseFilterPostTemporalSharpeningStrength = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
const WatermarkingStrength = {
    DEFAULT: "DEFAULT",
    LIGHTER: "LIGHTER",
    LIGHTEST: "LIGHTEST",
    STRONGER: "STRONGER",
    STRONGEST: "STRONGEST",
};
const TimecodeBurninPosition = {
    BOTTOM_CENTER: "BOTTOM_CENTER",
    BOTTOM_LEFT: "BOTTOM_LEFT",
    BOTTOM_RIGHT: "BOTTOM_RIGHT",
    MIDDLE_CENTER: "MIDDLE_CENTER",
    MIDDLE_LEFT: "MIDDLE_LEFT",
    MIDDLE_RIGHT: "MIDDLE_RIGHT",
    TOP_CENTER: "TOP_CENTER",
    TOP_LEFT: "TOP_LEFT",
    TOP_RIGHT: "TOP_RIGHT",
};
const TimecodeSource = {
    EMBEDDED: "EMBEDDED",
    SPECIFIEDSTART: "SPECIFIEDSTART",
    ZEROBASED: "ZEROBASED",
};
const ShareStatus = {
    INITIATED: "INITIATED",
    NOT_SHARED: "NOT_SHARED",
    SHARED: "SHARED",
};
const SimulateReservedQueue = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const JobStatus = {
    CANCELED: "CANCELED",
    COMPLETE: "COMPLETE",
    ERROR: "ERROR",
    PROGRESSING: "PROGRESSING",
    SUBMITTED: "SUBMITTED",
};
const StatusUpdateInterval = {
    SECONDS_10: "SECONDS_10",
    SECONDS_12: "SECONDS_12",
    SECONDS_120: "SECONDS_120",
    SECONDS_15: "SECONDS_15",
    SECONDS_180: "SECONDS_180",
    SECONDS_20: "SECONDS_20",
    SECONDS_240: "SECONDS_240",
    SECONDS_30: "SECONDS_30",
    SECONDS_300: "SECONDS_300",
    SECONDS_360: "SECONDS_360",
    SECONDS_420: "SECONDS_420",
    SECONDS_480: "SECONDS_480",
    SECONDS_540: "SECONDS_540",
    SECONDS_60: "SECONDS_60",
    SECONDS_600: "SECONDS_600",
};
const JobsQueryFilterKey = {
    audioCodec: "audioCodec",
    fileInput: "fileInput",
    jobEngineVersionRequested: "jobEngineVersionRequested",
    jobEngineVersionUsed: "jobEngineVersionUsed",
    queue: "queue",
    status: "status",
    videoCodec: "videoCodec",
};
const Type = {
    CUSTOM: "CUSTOM",
    SYSTEM: "SYSTEM",
};
const Format = {
    matroska: "matroska",
    mp4: "mp4",
    mxf: "mxf",
    quicktime: "quicktime",
    wave: "wave",
    webm: "webm",
};
const Codec = {
    AAC: "AAC",
    AC3: "AC3",
    AV1: "AV1",
    AVC: "AVC",
    C608: "C608",
    C708: "C708",
    EAC3: "EAC3",
    FLAC: "FLAC",
    HEVC: "HEVC",
    JPEG2000: "JPEG2000",
    MJPEG: "MJPEG",
    MP3: "MP3",
    MP4V: "MP4V",
    MPEG1: "MPEG1",
    MPEG2: "MPEG2",
    OPUS: "OPUS",
    PCM: "PCM",
    PRORES: "PRORES",
    QTRLE: "QTRLE",
    THEORA: "THEORA",
    UNKNOWN: "UNKNOWN",
    VFW: "VFW",
    VORBIS: "VORBIS",
    VP8: "VP8",
    VP9: "VP9",
    WEBVTT: "WEBVTT",
};
const TrackType = {
    audio: "audio",
    data: "data",
    video: "video",
};
const ColorPrimaries = {
    EBU_3213_E: "EBU_3213_E",
    GENERIC_FILM: "GENERIC_FILM",
    IPT: "IPT",
    ITU_2020: "ITU_2020",
    ITU_470BG: "ITU_470BG",
    ITU_470M: "ITU_470M",
    ITU_709: "ITU_709",
    LAST: "LAST",
    RESERVED: "RESERVED",
    SMPTE_170M: "SMPTE_170M",
    SMPTE_2067XYZ: "SMPTE_2067XYZ",
    SMPTE_240M: "SMPTE_240M",
    SMPTE_428_1: "SMPTE_428_1",
    SMPTE_431_2: "SMPTE_431_2",
    SMPTE_EG_432_1: "SMPTE_EG_432_1",
    UNSPECIFIED: "UNSPECIFIED",
};
const MatrixCoefficients = {
    CD_CL: "CD_CL",
    CD_NCL: "CD_NCL",
    EBU3213: "EBU3213",
    FCC: "FCC",
    IPT: "IPT",
    ITU_2020_CL: "ITU_2020_CL",
    ITU_2020_NCL: "ITU_2020_NCL",
    ITU_2100ICtCp: "ITU_2100ICtCp",
    ITU_470BG: "ITU_470BG",
    ITU_709: "ITU_709",
    LAST: "LAST",
    RESERVED: "RESERVED",
    RGB: "RGB",
    SMPTE_170M: "SMPTE_170M",
    SMPTE_2085: "SMPTE_2085",
    SMPTE_240M: "SMPTE_240M",
    UNSPECIFIED: "UNSPECIFIED",
    YCgCo: "YCgCo",
};
const TransferCharacteristics = {
    ARIB_B67: "ARIB_B67",
    IEC_61966_2_1: "IEC_61966_2_1",
    IEC_61966_2_4: "IEC_61966_2_4",
    ITU_1361: "ITU_1361",
    ITU_2020_10bit: "ITU_2020_10bit",
    ITU_2020_12bit: "ITU_2020_12bit",
    ITU_470BG: "ITU_470BG",
    ITU_470M: "ITU_470M",
    ITU_709: "ITU_709",
    LAST: "LAST",
    LINEAR: "LINEAR",
    LOC10_2_5: "LOC10_2_5",
    LOG10_2: "LOG10_2",
    RESERVED: "RESERVED",
    SMPTE_170M: "SMPTE_170M",
    SMPTE_2084: "SMPTE_2084",
    SMPTE_240M: "SMPTE_240M",
    SMPTE_428_1: "SMPTE_428_1",
    UNSPECIFIED: "UNSPECIFIED",
};
const PricingPlan = {
    ON_DEMAND: "ON_DEMAND",
    RESERVED: "RESERVED",
};
const Commitment = {
    ONE_YEAR: "ONE_YEAR",
};
const RenewalType = {
    AUTO_RENEW: "AUTO_RENEW",
    EXPIRE: "EXPIRE",
};
const ReservationPlanStatus = {
    ACTIVE: "ACTIVE",
    EXPIRED: "EXPIRED",
};
const QueueStatus = {
    ACTIVE: "ACTIVE",
    PAUSED: "PAUSED",
};
const DescribeEndpointsMode = {
    DEFAULT: "DEFAULT",
    GET_ONLY: "GET_ONLY",
};
const JobsQueryStatus = {
    COMPLETE: "COMPLETE",
    ERROR: "ERROR",
    PROGRESSING: "PROGRESSING",
    SUBMITTED: "SUBMITTED",
};
const InputPolicy = {
    ALLOWED: "ALLOWED",
    DISALLOWED: "DISALLOWED",
};
const JobTemplateListBy = {
    CREATION_DATE: "CREATION_DATE",
    NAME: "NAME",
    SYSTEM: "SYSTEM",
};
const Order = {
    ASCENDING: "ASCENDING",
    DESCENDING: "DESCENDING",
};
const PresetListBy = {
    CREATION_DATE: "CREATION_DATE",
    NAME: "NAME",
    SYSTEM: "SYSTEM",
};
const QueueListBy = {
    CREATION_DATE: "CREATION_DATE",
    NAME: "NAME",
};

exports.$Command = smithyClient.Command;
exports.__Client = smithyClient.Client;
exports.MediaConvertServiceException = MediaConvertServiceException.MediaConvertServiceException;
exports.AacAudioDescriptionBroadcasterMix = AacAudioDescriptionBroadcasterMix;
exports.AacCodecProfile = AacCodecProfile;
exports.AacCodingMode = AacCodingMode;
exports.AacLoudnessMeasurementMode = AacLoudnessMeasurementMode;
exports.AacRateControlMode = AacRateControlMode;
exports.AacRawFormat = AacRawFormat;
exports.AacSpecification = AacSpecification;
exports.AacVbrQuality = AacVbrQuality;
exports.Ac3BitstreamMode = Ac3BitstreamMode;
exports.Ac3CodingMode = Ac3CodingMode;
exports.Ac3DynamicRangeCompressionLine = Ac3DynamicRangeCompressionLine;
exports.Ac3DynamicRangeCompressionProfile = Ac3DynamicRangeCompressionProfile;
exports.Ac3DynamicRangeCompressionRf = Ac3DynamicRangeCompressionRf;
exports.Ac3LfeFilter = Ac3LfeFilter;
exports.Ac3MetadataControl = Ac3MetadataControl;
exports.AccelerationMode = AccelerationMode;
exports.AccelerationStatus = AccelerationStatus;
exports.AdvancedInputFilter = AdvancedInputFilter;
exports.AdvancedInputFilterAddTexture = AdvancedInputFilterAddTexture;
exports.AdvancedInputFilterSharpen = AdvancedInputFilterSharpen;
exports.AfdSignaling = AfdSignaling;
exports.AlphaBehavior = AlphaBehavior;
exports.AncillaryConvert608To708 = AncillaryConvert608To708;
exports.AncillaryTerminateCaptions = AncillaryTerminateCaptions;
exports.AntiAlias = AntiAlias;
exports.AssociateCertificateCommand = AssociateCertificateCommand;
exports.AudioChannelTag = AudioChannelTag;
exports.AudioCodec = AudioCodec;
exports.AudioDefaultSelection = AudioDefaultSelection;
exports.AudioDurationCorrection = AudioDurationCorrection;
exports.AudioLanguageCodeControl = AudioLanguageCodeControl;
exports.AudioNormalizationAlgorithm = AudioNormalizationAlgorithm;
exports.AudioNormalizationAlgorithmControl = AudioNormalizationAlgorithmControl;
exports.AudioNormalizationLoudnessLogging = AudioNormalizationLoudnessLogging;
exports.AudioNormalizationPeakCalculation = AudioNormalizationPeakCalculation;
exports.AudioSelectorType = AudioSelectorType;
exports.AudioTypeControl = AudioTypeControl;
exports.Av1AdaptiveQuantization = Av1AdaptiveQuantization;
exports.Av1BitDepth = Av1BitDepth;
exports.Av1FilmGrainSynthesis = Av1FilmGrainSynthesis;
exports.Av1FramerateControl = Av1FramerateControl;
exports.Av1FramerateConversionAlgorithm = Av1FramerateConversionAlgorithm;
exports.Av1RateControlMode = Av1RateControlMode;
exports.Av1SpatialAdaptiveQuantization = Av1SpatialAdaptiveQuantization;
exports.AvcIntraClass = AvcIntraClass;
exports.AvcIntraFramerateControl = AvcIntraFramerateControl;
exports.AvcIntraFramerateConversionAlgorithm = AvcIntraFramerateConversionAlgorithm;
exports.AvcIntraInterlaceMode = AvcIntraInterlaceMode;
exports.AvcIntraScanTypeConversionMode = AvcIntraScanTypeConversionMode;
exports.AvcIntraSlowPal = AvcIntraSlowPal;
exports.AvcIntraTelecine = AvcIntraTelecine;
exports.AvcIntraUhdQualityTuningLevel = AvcIntraUhdQualityTuningLevel;
exports.BandwidthReductionFilterSharpening = BandwidthReductionFilterSharpening;
exports.BandwidthReductionFilterStrength = BandwidthReductionFilterStrength;
exports.BillingTagsSource = BillingTagsSource;
exports.BurnInSubtitleStylePassthrough = BurnInSubtitleStylePassthrough;
exports.BurninSubtitleAlignment = BurninSubtitleAlignment;
exports.BurninSubtitleApplyFontColor = BurninSubtitleApplyFontColor;
exports.BurninSubtitleBackgroundColor = BurninSubtitleBackgroundColor;
exports.BurninSubtitleFallbackFont = BurninSubtitleFallbackFont;
exports.BurninSubtitleFontColor = BurninSubtitleFontColor;
exports.BurninSubtitleOutlineColor = BurninSubtitleOutlineColor;
exports.BurninSubtitleShadowColor = BurninSubtitleShadowColor;
exports.BurninSubtitleTeletextSpacing = BurninSubtitleTeletextSpacing;
exports.CancelJobCommand = CancelJobCommand;
exports.CaptionDestinationType = CaptionDestinationType;
exports.CaptionSourceByteRateLimit = CaptionSourceByteRateLimit;
exports.CaptionSourceConvertPaintOnToPopOn = CaptionSourceConvertPaintOnToPopOn;
exports.CaptionSourceType = CaptionSourceType;
exports.CaptionSourceUpconvertSTLToTeletext = CaptionSourceUpconvertSTLToTeletext;
exports.ChromaPositionMode = ChromaPositionMode;
exports.CmafClientCache = CmafClientCache;
exports.CmafCodecSpecification = CmafCodecSpecification;
exports.CmafEncryptionType = CmafEncryptionType;
exports.CmafImageBasedTrickPlay = CmafImageBasedTrickPlay;
exports.CmafInitializationVectorInManifest = CmafInitializationVectorInManifest;
exports.CmafIntervalCadence = CmafIntervalCadence;
exports.CmafKeyProviderType = CmafKeyProviderType;
exports.CmafManifestCompression = CmafManifestCompression;
exports.CmafManifestDurationFormat = CmafManifestDurationFormat;
exports.CmafMpdManifestBandwidthType = CmafMpdManifestBandwidthType;
exports.CmafMpdProfile = CmafMpdProfile;
exports.CmafPtsOffsetHandlingForBFrames = CmafPtsOffsetHandlingForBFrames;
exports.CmafSegmentControl = CmafSegmentControl;
exports.CmafSegmentLengthControl = CmafSegmentLengthControl;
exports.CmafStreamInfResolution = CmafStreamInfResolution;
exports.CmafTargetDurationCompatibilityMode = CmafTargetDurationCompatibilityMode;
exports.CmafVideoCompositionOffsets = CmafVideoCompositionOffsets;
exports.CmafWriteDASHManifest = CmafWriteDASHManifest;
exports.CmafWriteHLSManifest = CmafWriteHLSManifest;
exports.CmafWriteSegmentTimelineInRepresentation = CmafWriteSegmentTimelineInRepresentation;
exports.CmfcAudioDuration = CmfcAudioDuration;
exports.CmfcAudioTrackType = CmfcAudioTrackType;
exports.CmfcC2paManifest = CmfcC2paManifest;
exports.CmfcDescriptiveVideoServiceFlag = CmfcDescriptiveVideoServiceFlag;
exports.CmfcIFrameOnlyManifest = CmfcIFrameOnlyManifest;
exports.CmfcKlvMetadata = CmfcKlvMetadata;
exports.CmfcManifestMetadataSignaling = CmfcManifestMetadataSignaling;
exports.CmfcScte35Esam = CmfcScte35Esam;
exports.CmfcScte35Source = CmfcScte35Source;
exports.CmfcTimedMetadata = CmfcTimedMetadata;
exports.CmfcTimedMetadataBoxVersion = CmfcTimedMetadataBoxVersion;
exports.Codec = Codec;
exports.ColorMetadata = ColorMetadata;
exports.ColorPrimaries = ColorPrimaries;
exports.ColorSpace = ColorSpace;
exports.ColorSpaceConversion = ColorSpaceConversion;
exports.ColorSpaceUsage = ColorSpaceUsage;
exports.Commitment = Commitment;
exports.ContainerType = ContainerType;
exports.CopyProtectionAction = CopyProtectionAction;
exports.CreateJobCommand = CreateJobCommand;
exports.CreateJobTemplateCommand = CreateJobTemplateCommand;
exports.CreatePresetCommand = CreatePresetCommand;
exports.CreateQueueCommand = CreateQueueCommand;
exports.CreateResourceShareCommand = CreateResourceShareCommand;
exports.DashIsoGroupAudioChannelConfigSchemeIdUri = DashIsoGroupAudioChannelConfigSchemeIdUri;
exports.DashIsoHbbtvCompliance = DashIsoHbbtvCompliance;
exports.DashIsoImageBasedTrickPlay = DashIsoImageBasedTrickPlay;
exports.DashIsoIntervalCadence = DashIsoIntervalCadence;
exports.DashIsoMpdManifestBandwidthType = DashIsoMpdManifestBandwidthType;
exports.DashIsoMpdProfile = DashIsoMpdProfile;
exports.DashIsoPlaybackDeviceCompatibility = DashIsoPlaybackDeviceCompatibility;
exports.DashIsoPtsOffsetHandlingForBFrames = DashIsoPtsOffsetHandlingForBFrames;
exports.DashIsoSegmentControl = DashIsoSegmentControl;
exports.DashIsoSegmentLengthControl = DashIsoSegmentLengthControl;
exports.DashIsoVideoCompositionOffsets = DashIsoVideoCompositionOffsets;
exports.DashIsoWriteSegmentTimelineInRepresentation = DashIsoWriteSegmentTimelineInRepresentation;
exports.DashManifestStyle = DashManifestStyle;
exports.DecryptionMode = DecryptionMode;
exports.DeinterlaceAlgorithm = DeinterlaceAlgorithm;
exports.DeinterlacerControl = DeinterlacerControl;
exports.DeinterlacerMode = DeinterlacerMode;
exports.DeleteJobTemplateCommand = DeleteJobTemplateCommand;
exports.DeletePolicyCommand = DeletePolicyCommand;
exports.DeletePresetCommand = DeletePresetCommand;
exports.DeleteQueueCommand = DeleteQueueCommand;
exports.DescribeEndpointsCommand = DescribeEndpointsCommand;
exports.DescribeEndpointsMode = DescribeEndpointsMode;
exports.DisassociateCertificateCommand = DisassociateCertificateCommand;
exports.DolbyVisionCompatibility = DolbyVisionCompatibility;
exports.DolbyVisionLevel6Mode = DolbyVisionLevel6Mode;
exports.DolbyVisionMapping = DolbyVisionMapping;
exports.DolbyVisionProfile = DolbyVisionProfile;
exports.DropFrameTimecode = DropFrameTimecode;
exports.DvbSubSubtitleFallbackFont = DvbSubSubtitleFallbackFont;
exports.DvbSubtitleAlignment = DvbSubtitleAlignment;
exports.DvbSubtitleApplyFontColor = DvbSubtitleApplyFontColor;
exports.DvbSubtitleBackgroundColor = DvbSubtitleBackgroundColor;
exports.DvbSubtitleFontColor = DvbSubtitleFontColor;
exports.DvbSubtitleOutlineColor = DvbSubtitleOutlineColor;
exports.DvbSubtitleShadowColor = DvbSubtitleShadowColor;
exports.DvbSubtitleStylePassthrough = DvbSubtitleStylePassthrough;
exports.DvbSubtitleTeletextSpacing = DvbSubtitleTeletextSpacing;
exports.DvbSubtitlingType = DvbSubtitlingType;
exports.DvbddsHandling = DvbddsHandling;
exports.DynamicAudioSelectorType = DynamicAudioSelectorType;
exports.Eac3AtmosBitstreamMode = Eac3AtmosBitstreamMode;
exports.Eac3AtmosCodingMode = Eac3AtmosCodingMode;
exports.Eac3AtmosDialogueIntelligence = Eac3AtmosDialogueIntelligence;
exports.Eac3AtmosDownmixControl = Eac3AtmosDownmixControl;
exports.Eac3AtmosDynamicRangeCompressionLine = Eac3AtmosDynamicRangeCompressionLine;
exports.Eac3AtmosDynamicRangeCompressionRf = Eac3AtmosDynamicRangeCompressionRf;
exports.Eac3AtmosDynamicRangeControl = Eac3AtmosDynamicRangeControl;
exports.Eac3AtmosMeteringMode = Eac3AtmosMeteringMode;
exports.Eac3AtmosStereoDownmix = Eac3AtmosStereoDownmix;
exports.Eac3AtmosSurroundExMode = Eac3AtmosSurroundExMode;
exports.Eac3AttenuationControl = Eac3AttenuationControl;
exports.Eac3BitstreamMode = Eac3BitstreamMode;
exports.Eac3CodingMode = Eac3CodingMode;
exports.Eac3DcFilter = Eac3DcFilter;
exports.Eac3DynamicRangeCompressionLine = Eac3DynamicRangeCompressionLine;
exports.Eac3DynamicRangeCompressionRf = Eac3DynamicRangeCompressionRf;
exports.Eac3LfeControl = Eac3LfeControl;
exports.Eac3LfeFilter = Eac3LfeFilter;
exports.Eac3MetadataControl = Eac3MetadataControl;
exports.Eac3PassthroughControl = Eac3PassthroughControl;
exports.Eac3PhaseControl = Eac3PhaseControl;
exports.Eac3StereoDownmix = Eac3StereoDownmix;
exports.Eac3SurroundExMode = Eac3SurroundExMode;
exports.Eac3SurroundMode = Eac3SurroundMode;
exports.EmbeddedConvert608To708 = EmbeddedConvert608To708;
exports.EmbeddedTerminateCaptions = EmbeddedTerminateCaptions;
exports.EmbeddedTimecodeOverride = EmbeddedTimecodeOverride;
exports.F4vMoovPlacement = F4vMoovPlacement;
exports.FileSourceConvert608To708 = FileSourceConvert608To708;
exports.FileSourceTimeDeltaUnits = FileSourceTimeDeltaUnits;
exports.FontScript = FontScript;
exports.Format = Format;
exports.FrameControl = FrameControl;
exports.FrameMetricType = FrameMetricType;
exports.GetJobCommand = GetJobCommand;
exports.GetJobTemplateCommand = GetJobTemplateCommand;
exports.GetJobsQueryResultsCommand = GetJobsQueryResultsCommand;
exports.GetPolicyCommand = GetPolicyCommand;
exports.GetPresetCommand = GetPresetCommand;
exports.GetQueueCommand = GetQueueCommand;
exports.GifFramerateControl = GifFramerateControl;
exports.GifFramerateConversionAlgorithm = GifFramerateConversionAlgorithm;
exports.H264AdaptiveQuantization = H264AdaptiveQuantization;
exports.H264CodecLevel = H264CodecLevel;
exports.H264CodecProfile = H264CodecProfile;
exports.H264DynamicSubGop = H264DynamicSubGop;
exports.H264EndOfStreamMarkers = H264EndOfStreamMarkers;
exports.H264EntropyEncoding = H264EntropyEncoding;
exports.H264FieldEncoding = H264FieldEncoding;
exports.H264FlickerAdaptiveQuantization = H264FlickerAdaptiveQuantization;
exports.H264FramerateControl = H264FramerateControl;
exports.H264FramerateConversionAlgorithm = H264FramerateConversionAlgorithm;
exports.H264GopBReference = H264GopBReference;
exports.H264GopSizeUnits = H264GopSizeUnits;
exports.H264InterlaceMode = H264InterlaceMode;
exports.H264ParControl = H264ParControl;
exports.H264QualityTuningLevel = H264QualityTuningLevel;
exports.H264RateControlMode = H264RateControlMode;
exports.H264RepeatPps = H264RepeatPps;
exports.H264SaliencyAwareEncoding = H264SaliencyAwareEncoding;
exports.H264ScanTypeConversionMode = H264ScanTypeConversionMode;
exports.H264SceneChangeDetect = H264SceneChangeDetect;
exports.H264SlowPal = H264SlowPal;
exports.H264SpatialAdaptiveQuantization = H264SpatialAdaptiveQuantization;
exports.H264Syntax = H264Syntax;
exports.H264Telecine = H264Telecine;
exports.H264TemporalAdaptiveQuantization = H264TemporalAdaptiveQuantization;
exports.H264UnregisteredSeiTimecode = H264UnregisteredSeiTimecode;
exports.H264WriteMp4PackagingType = H264WriteMp4PackagingType;
exports.H265AdaptiveQuantization = H265AdaptiveQuantization;
exports.H265AlternateTransferFunctionSei = H265AlternateTransferFunctionSei;
exports.H265CodecLevel = H265CodecLevel;
exports.H265CodecProfile = H265CodecProfile;
exports.H265Deblocking = H265Deblocking;
exports.H265DynamicSubGop = H265DynamicSubGop;
exports.H265EndOfStreamMarkers = H265EndOfStreamMarkers;
exports.H265FlickerAdaptiveQuantization = H265FlickerAdaptiveQuantization;
exports.H265FramerateControl = H265FramerateControl;
exports.H265FramerateConversionAlgorithm = H265FramerateConversionAlgorithm;
exports.H265GopBReference = H265GopBReference;
exports.H265GopSizeUnits = H265GopSizeUnits;
exports.H265InterlaceMode = H265InterlaceMode;
exports.H265MvOverPictureBoundaries = H265MvOverPictureBoundaries;
exports.H265MvTemporalPredictor = H265MvTemporalPredictor;
exports.H265ParControl = H265ParControl;
exports.H265QualityTuningLevel = H265QualityTuningLevel;
exports.H265RateControlMode = H265RateControlMode;
exports.H265SampleAdaptiveOffsetFilterMode = H265SampleAdaptiveOffsetFilterMode;
exports.H265ScanTypeConversionMode = H265ScanTypeConversionMode;
exports.H265SceneChangeDetect = H265SceneChangeDetect;
exports.H265SlowPal = H265SlowPal;
exports.H265SpatialAdaptiveQuantization = H265SpatialAdaptiveQuantization;
exports.H265Telecine = H265Telecine;
exports.H265TemporalAdaptiveQuantization = H265TemporalAdaptiveQuantization;
exports.H265TemporalIds = H265TemporalIds;
exports.H265TilePadding = H265TilePadding;
exports.H265Tiles = H265Tiles;
exports.H265TreeBlockSize = H265TreeBlockSize;
exports.H265UnregisteredSeiTimecode = H265UnregisteredSeiTimecode;
exports.H265WriteMp4PackagingType = H265WriteMp4PackagingType;
exports.HDRToSDRToneMapper = HDRToSDRToneMapper;
exports.HlsAdMarkers = HlsAdMarkers;
exports.HlsAudioOnlyContainer = HlsAudioOnlyContainer;
exports.HlsAudioOnlyHeader = HlsAudioOnlyHeader;
exports.HlsAudioTrackType = HlsAudioTrackType;
exports.HlsCaptionLanguageSetting = HlsCaptionLanguageSetting;
exports.HlsCaptionSegmentLengthControl = HlsCaptionSegmentLengthControl;
exports.HlsClientCache = HlsClientCache;
exports.HlsCodecSpecification = HlsCodecSpecification;
exports.HlsDescriptiveVideoServiceFlag = HlsDescriptiveVideoServiceFlag;
exports.HlsDirectoryStructure = HlsDirectoryStructure;
exports.HlsEncryptionType = HlsEncryptionType;
exports.HlsIFrameOnlyManifest = HlsIFrameOnlyManifest;
exports.HlsImageBasedTrickPlay = HlsImageBasedTrickPlay;
exports.HlsInitializationVectorInManifest = HlsInitializationVectorInManifest;
exports.HlsIntervalCadence = HlsIntervalCadence;
exports.HlsKeyProviderType = HlsKeyProviderType;
exports.HlsManifestCompression = HlsManifestCompression;
exports.HlsManifestDurationFormat = HlsManifestDurationFormat;
exports.HlsOfflineEncrypted = HlsOfflineEncrypted;
exports.HlsOutputSelection = HlsOutputSelection;
exports.HlsProgramDateTime = HlsProgramDateTime;
exports.HlsProgressiveWriteHlsManifest = HlsProgressiveWriteHlsManifest;
exports.HlsSegmentControl = HlsSegmentControl;
exports.HlsSegmentLengthControl = HlsSegmentLengthControl;
exports.HlsStreamInfResolution = HlsStreamInfResolution;
exports.HlsTargetDurationCompatibilityMode = HlsTargetDurationCompatibilityMode;
exports.HlsTimedMetadataId3Frame = HlsTimedMetadataId3Frame;
exports.ImscAccessibilitySubs = ImscAccessibilitySubs;
exports.ImscStylePassthrough = ImscStylePassthrough;
exports.InputDeblockFilter = InputDeblockFilter;
exports.InputDenoiseFilter = InputDenoiseFilter;
exports.InputFilterEnable = InputFilterEnable;
exports.InputPolicy = InputPolicy;
exports.InputPsiControl = InputPsiControl;
exports.InputRotate = InputRotate;
exports.InputSampleRange = InputSampleRange;
exports.InputScanType = InputScanType;
exports.InputTimecodeSource = InputTimecodeSource;
exports.JobPhase = JobPhase;
exports.JobStatus = JobStatus;
exports.JobTemplateListBy = JobTemplateListBy;
exports.JobsQueryFilterKey = JobsQueryFilterKey;
exports.JobsQueryStatus = JobsQueryStatus;
exports.LanguageCode = LanguageCode;
exports.ListJobTemplatesCommand = ListJobTemplatesCommand;
exports.ListJobsCommand = ListJobsCommand;
exports.ListPresetsCommand = ListPresetsCommand;
exports.ListQueuesCommand = ListQueuesCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListVersionsCommand = ListVersionsCommand;
exports.M2tsAudioBufferModel = M2tsAudioBufferModel;
exports.M2tsAudioDuration = M2tsAudioDuration;
exports.M2tsBufferModel = M2tsBufferModel;
exports.M2tsDataPtsControl = M2tsDataPtsControl;
exports.M2tsEbpAudioInterval = M2tsEbpAudioInterval;
exports.M2tsEbpPlacement = M2tsEbpPlacement;
exports.M2tsEsRateInPes = M2tsEsRateInPes;
exports.M2tsForceTsVideoEbpOrder = M2tsForceTsVideoEbpOrder;
exports.M2tsKlvMetadata = M2tsKlvMetadata;
exports.M2tsNielsenId3 = M2tsNielsenId3;
exports.M2tsPcrControl = M2tsPcrControl;
exports.M2tsPreventBufferUnderflow = M2tsPreventBufferUnderflow;
exports.M2tsRateMode = M2tsRateMode;
exports.M2tsScte35Source = M2tsScte35Source;
exports.M2tsSegmentationMarkers = M2tsSegmentationMarkers;
exports.M2tsSegmentationStyle = M2tsSegmentationStyle;
exports.M3u8AudioDuration = M3u8AudioDuration;
exports.M3u8DataPtsControl = M3u8DataPtsControl;
exports.M3u8NielsenId3 = M3u8NielsenId3;
exports.M3u8PcrControl = M3u8PcrControl;
exports.M3u8Scte35Source = M3u8Scte35Source;
exports.MatrixCoefficients = MatrixCoefficients;
exports.MediaConvert = MediaConvert;
exports.MediaConvertClient = MediaConvertClient;
exports.MotionImageInsertionMode = MotionImageInsertionMode;
exports.MotionImagePlayback = MotionImagePlayback;
exports.MovClapAtom = MovClapAtom;
exports.MovCslgAtom = MovCslgAtom;
exports.MovMpeg2FourCCControl = MovMpeg2FourCCControl;
exports.MovPaddingControl = MovPaddingControl;
exports.MovReference = MovReference;
exports.Mp2AudioDescriptionMix = Mp2AudioDescriptionMix;
exports.Mp3RateControlMode = Mp3RateControlMode;
exports.Mp4C2paManifest = Mp4C2paManifest;
exports.Mp4CslgAtom = Mp4CslgAtom;
exports.Mp4FreeSpaceBox = Mp4FreeSpaceBox;
exports.Mp4MoovPlacement = Mp4MoovPlacement;
exports.MpdAccessibilityCaptionHints = MpdAccessibilityCaptionHints;
exports.MpdAudioDuration = MpdAudioDuration;
exports.MpdC2paManifest = MpdC2paManifest;
exports.MpdCaptionContainerType = MpdCaptionContainerType;
exports.MpdKlvMetadata = MpdKlvMetadata;
exports.MpdManifestMetadataSignaling = MpdManifestMetadataSignaling;
exports.MpdScte35Esam = MpdScte35Esam;
exports.MpdScte35Source = MpdScte35Source;
exports.MpdTimedMetadata = MpdTimedMetadata;
exports.MpdTimedMetadataBoxVersion = MpdTimedMetadataBoxVersion;
exports.Mpeg2AdaptiveQuantization = Mpeg2AdaptiveQuantization;
exports.Mpeg2CodecLevel = Mpeg2CodecLevel;
exports.Mpeg2CodecProfile = Mpeg2CodecProfile;
exports.Mpeg2DynamicSubGop = Mpeg2DynamicSubGop;
exports.Mpeg2FramerateControl = Mpeg2FramerateControl;
exports.Mpeg2FramerateConversionAlgorithm = Mpeg2FramerateConversionAlgorithm;
exports.Mpeg2GopSizeUnits = Mpeg2GopSizeUnits;
exports.Mpeg2InterlaceMode = Mpeg2InterlaceMode;
exports.Mpeg2IntraDcPrecision = Mpeg2IntraDcPrecision;
exports.Mpeg2ParControl = Mpeg2ParControl;
exports.Mpeg2QualityTuningLevel = Mpeg2QualityTuningLevel;
exports.Mpeg2RateControlMode = Mpeg2RateControlMode;
exports.Mpeg2ScanTypeConversionMode = Mpeg2ScanTypeConversionMode;
exports.Mpeg2SceneChangeDetect = Mpeg2SceneChangeDetect;
exports.Mpeg2SlowPal = Mpeg2SlowPal;
exports.Mpeg2SpatialAdaptiveQuantization = Mpeg2SpatialAdaptiveQuantization;
exports.Mpeg2Syntax = Mpeg2Syntax;
exports.Mpeg2Telecine = Mpeg2Telecine;
exports.Mpeg2TemporalAdaptiveQuantization = Mpeg2TemporalAdaptiveQuantization;
exports.MsSmoothAudioDeduplication = MsSmoothAudioDeduplication;
exports.MsSmoothFragmentLengthControl = MsSmoothFragmentLengthControl;
exports.MsSmoothManifestEncoding = MsSmoothManifestEncoding;
exports.MxfAfdSignaling = MxfAfdSignaling;
exports.MxfProfile = MxfProfile;
exports.MxfUncompressedAudioWrapping = MxfUncompressedAudioWrapping;
exports.MxfXavcDurationMode = MxfXavcDurationMode;
exports.NielsenActiveWatermarkProcessType = NielsenActiveWatermarkProcessType;
exports.NielsenSourceWatermarkStatusType = NielsenSourceWatermarkStatusType;
exports.NielsenUniqueTicPerAudioTrackType = NielsenUniqueTicPerAudioTrackType;
exports.NoiseFilterPostTemporalSharpening = NoiseFilterPostTemporalSharpening;
exports.NoiseFilterPostTemporalSharpeningStrength = NoiseFilterPostTemporalSharpeningStrength;
exports.NoiseReducerFilter = NoiseReducerFilter;
exports.Order = Order;
exports.OutputGroupType = OutputGroupType;
exports.OutputSdt = OutputSdt;
exports.PadVideo = PadVideo;
exports.PresetListBy = PresetListBy;
exports.PresetSpeke20Audio = PresetSpeke20Audio;
exports.PresetSpeke20Video = PresetSpeke20Video;
exports.PricingPlan = PricingPlan;
exports.ProbeCommand = ProbeCommand;
exports.ProresChromaSampling = ProresChromaSampling;
exports.ProresCodecProfile = ProresCodecProfile;
exports.ProresFramerateControl = ProresFramerateControl;
exports.ProresFramerateConversionAlgorithm = ProresFramerateConversionAlgorithm;
exports.ProresInterlaceMode = ProresInterlaceMode;
exports.ProresParControl = ProresParControl;
exports.ProresScanTypeConversionMode = ProresScanTypeConversionMode;
exports.ProresSlowPal = ProresSlowPal;
exports.ProresTelecine = ProresTelecine;
exports.PutPolicyCommand = PutPolicyCommand;
exports.QueueListBy = QueueListBy;
exports.QueueStatus = QueueStatus;
exports.RemoveRubyReserveAttributes = RemoveRubyReserveAttributes;
exports.RenewalType = RenewalType;
exports.RequiredFlag = RequiredFlag;
exports.ReservationPlanStatus = ReservationPlanStatus;
exports.RespondToAfd = RespondToAfd;
exports.RuleType = RuleType;
exports.S3ObjectCannedAcl = S3ObjectCannedAcl;
exports.S3ServerSideEncryptionType = S3ServerSideEncryptionType;
exports.S3StorageClass = S3StorageClass;
exports.SampleRangeConversion = SampleRangeConversion;
exports.ScalingBehavior = ScalingBehavior;
exports.SccDestinationFramerate = SccDestinationFramerate;
exports.SearchJobsCommand = SearchJobsCommand;
exports.ShareStatus = ShareStatus;
exports.SimulateReservedQueue = SimulateReservedQueue;
exports.SlowPalPitchCorrection = SlowPalPitchCorrection;
exports.SrtStylePassthrough = SrtStylePassthrough;
exports.StartJobsQueryCommand = StartJobsQueryCommand;
exports.StatusUpdateInterval = StatusUpdateInterval;
exports.TagResourceCommand = TagResourceCommand;
exports.TamsGapHandling = TamsGapHandling;
exports.TeletextPageType = TeletextPageType;
exports.TimecodeBurninPosition = TimecodeBurninPosition;
exports.TimecodeSource = TimecodeSource;
exports.TimecodeTrack = TimecodeTrack;
exports.TimedMetadata = TimedMetadata;
exports.TrackType = TrackType;
exports.TransferCharacteristics = TransferCharacteristics;
exports.TsPtsOffset = TsPtsOffset;
exports.TtmlStylePassthrough = TtmlStylePassthrough;
exports.Type = Type;
exports.UncompressedFourcc = UncompressedFourcc;
exports.UncompressedFramerateControl = UncompressedFramerateControl;
exports.UncompressedFramerateConversionAlgorithm = UncompressedFramerateConversionAlgorithm;
exports.UncompressedInterlaceMode = UncompressedInterlaceMode;
exports.UncompressedScanTypeConversionMode = UncompressedScanTypeConversionMode;
exports.UncompressedSlowPal = UncompressedSlowPal;
exports.UncompressedTelecine = UncompressedTelecine;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateJobTemplateCommand = UpdateJobTemplateCommand;
exports.UpdatePresetCommand = UpdatePresetCommand;
exports.UpdateQueueCommand = UpdateQueueCommand;
exports.Vc3Class = Vc3Class;
exports.Vc3FramerateControl = Vc3FramerateControl;
exports.Vc3FramerateConversionAlgorithm = Vc3FramerateConversionAlgorithm;
exports.Vc3InterlaceMode = Vc3InterlaceMode;
exports.Vc3ScanTypeConversionMode = Vc3ScanTypeConversionMode;
exports.Vc3SlowPal = Vc3SlowPal;
exports.Vc3Telecine = Vc3Telecine;
exports.VchipAction = VchipAction;
exports.VideoCodec = VideoCodec;
exports.VideoOverlayPlayBackMode = VideoOverlayPlayBackMode;
exports.VideoOverlayUnit = VideoOverlayUnit;
exports.VideoSelectorMode = VideoSelectorMode;
exports.VideoSelectorType = VideoSelectorType;
exports.VideoTimecodeInsertion = VideoTimecodeInsertion;
exports.Vp8FramerateControl = Vp8FramerateControl;
exports.Vp8FramerateConversionAlgorithm = Vp8FramerateConversionAlgorithm;
exports.Vp8ParControl = Vp8ParControl;
exports.Vp8QualityTuningLevel = Vp8QualityTuningLevel;
exports.Vp8RateControlMode = Vp8RateControlMode;
exports.Vp9FramerateControl = Vp9FramerateControl;
exports.Vp9FramerateConversionAlgorithm = Vp9FramerateConversionAlgorithm;
exports.Vp9ParControl = Vp9ParControl;
exports.Vp9QualityTuningLevel = Vp9QualityTuningLevel;
exports.Vp9RateControlMode = Vp9RateControlMode;
exports.WatermarkingStrength = WatermarkingStrength;
exports.WavFormat = WavFormat;
exports.WebvttAccessibilitySubs = WebvttAccessibilitySubs;
exports.WebvttStylePassthrough = WebvttStylePassthrough;
exports.Xavc4kIntraCbgProfileClass = Xavc4kIntraCbgProfileClass;
exports.Xavc4kIntraVbrProfileClass = Xavc4kIntraVbrProfileClass;
exports.Xavc4kProfileBitrateClass = Xavc4kProfileBitrateClass;
exports.Xavc4kProfileCodecProfile = Xavc4kProfileCodecProfile;
exports.Xavc4kProfileQualityTuningLevel = Xavc4kProfileQualityTuningLevel;
exports.XavcAdaptiveQuantization = XavcAdaptiveQuantization;
exports.XavcEntropyEncoding = XavcEntropyEncoding;
exports.XavcFlickerAdaptiveQuantization = XavcFlickerAdaptiveQuantization;
exports.XavcFramerateControl = XavcFramerateControl;
exports.XavcFramerateConversionAlgorithm = XavcFramerateConversionAlgorithm;
exports.XavcGopBReference = XavcGopBReference;
exports.XavcHdIntraCbgProfileClass = XavcHdIntraCbgProfileClass;
exports.XavcHdProfileBitrateClass = XavcHdProfileBitrateClass;
exports.XavcHdProfileQualityTuningLevel = XavcHdProfileQualityTuningLevel;
exports.XavcHdProfileTelecine = XavcHdProfileTelecine;
exports.XavcInterlaceMode = XavcInterlaceMode;
exports.XavcProfile = XavcProfile;
exports.XavcSlowPal = XavcSlowPal;
exports.XavcSpatialAdaptiveQuantization = XavcSpatialAdaptiveQuantization;
exports.XavcTemporalAdaptiveQuantization = XavcTemporalAdaptiveQuantization;
exports.paginateDescribeEndpoints = paginateDescribeEndpoints;
exports.paginateListJobTemplates = paginateListJobTemplates;
exports.paginateListJobs = paginateListJobs;
exports.paginateListPresets = paginateListPresets;
exports.paginateListQueues = paginateListQueues;
exports.paginateListVersions = paginateListVersions;
exports.paginateSearchJobs = paginateSearchJobs;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
